// @rohit23061999/transaction-review-widget/dist/taxInsightDashboardWidget/taxInsightDashboard.js
"use client";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect, useState } from "react";
import api from "../api/api";
import { toast } from "react-toastify";
import { getAuthSessionToken } from "../helper/AuthHelper";
import ErrorWidget from "../helper/Error";
import TaxSummaryDashboard from "../helper/TaxSummaryDashboard";
const TaxInsightWidget = ({ userId, access_token, session_token, onError, }) => {
    const [error, setError] = useState("");
    const [isOpen, setIsOpen] = useState(false);
    const [data, setData] = useState();
    const [loading, setLoading] = useState(false);
    const handleApiCalling = (type) => __awaiter(void 0, void 0, void 0, function* () {
        var _a, _b, _c, _d;
        if (type) {
            setLoading(true);
        }
        try {
            const response = yield api.get(`/V2/api/tax-analysis/full-analysis?user_id=686cf76324795af55d4f5302`);
            console.log(response, "responseresponse");
            if ((response === null || response === void 0 ? void 0 : response.status) == 200) {
                setData(response === null || response === void 0 ? void 0 : response.data);
                setLoading(false);
            }
            else {
                setData([]);
                setLoading(false);
            }
        }
        catch (error) {
            setLoading(false);
            setData([]);
            console.error("API error:", (_b = (_a = error === null || error === void 0 ? void 0 : error.response) === null || _a === void 0 ? void 0 : _a.data) === null || _b === void 0 ? void 0 : _b.detail);
            toast.error(((_d = (_c = error === null || error === void 0 ? void 0 : error.response) === null || _c === void 0 ? void 0 : _c.data) === null || _d === void 0 ? void 0 : _d.detail) || "Failed to fetch tax return data", { toastId: "failed" });
        }
        finally {
            setLoading(false);
        }
    });
    useEffect(() => {
        if (access_token && session_token) {
            getAuthSessionToken({
                session_token,
                userId,
                onSuccess: (data) => {
                    handleApiCalling(true);
                    console.log("Auth token fetched, domain URLs:", data.domain_urls);
                },
                onError,
                setLoading,
                setError,
            });
        }
    }, [access_token, session_token]);
    return (_jsxs(_Fragment, { children: [_jsx("style", { children: `
        .custom-scrollbar::-webkit-scrollbar {
          width: 8px;
        }

        .custom-scrollbar::-webkit-scrollbar-track {
          background: transparent;
          border-radius: 10px;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb {
          background-color: #ccc;
          border-radius: 10px;
          border: 2px solid transparent;
          background-clip: content-box;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb:hover {
          background-color: #999;
        }

        .custom-scrollbar {
          scrollbar-width: thin;
          scrollbar-color: #ccc transparent;
        }
      ` }), _jsx("div", Object.assign({ style: {
                    position: "fixed",
                    bottom: "20px",
                    right: "20px",
                    width: "60px",
                    height: "60px",
                    backgroundColor: "#007bff",
                    borderRadius: "50%",
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                    cursor: "pointer",
                    zIndex: 9999,
                    color: "#fff",
                    fontSize: "28px",
                    boxShadow: "0 4px 5px rgba(0, 0, 0, 0.3)",
                }, onClick: () => setIsOpen(!isOpen), title: "Open Auth Widget" }, { children: "\uD83D\uDCAC" })), isOpen && (_jsx("div", Object.assign({ className: "custom-scrollbar", style: {
                    position: "fixed",
                    bottom: "90px",
                    right: "20px",
                    width: "68rem",
                    maxHeight: "75vh",
                    zIndex: 9998,
                    padding: "0",
                    overflowY: "auto",
                } }, { children: _jsx("div", Object.assign({ style: {
                        maxWidth: 1200,
                        margin: "auto",
                        border: "1px solid #ccc",
                        borderRadius: 16,
                        overflow: "hidden",
                        backgroundColor: "#ffffff",
                    } }, { children: error ? (_jsx("div", Object.assign({ style: {
                            padding: "0",
                            minHeight: "500px",
                            paddingTop: "200px",
                        } }, { children: _jsx(ErrorWidget, { message: error, onClose: () => setIsOpen(false) }) }))) : (_jsxs(_Fragment, { children: [_jsx("div", Object.assign({ style: {
                                    background: "#f1f7fb",
                                    padding: "16px 28px",
                                    fontSize: 20,
                                    fontWeight: "bold",
                                } }, { children: "Tax Insights Dashboard" })), _jsxs("div", Object.assign({ style: { padding: "10px 24px" } }, { children: [_jsx("p", Object.assign({ style: {
                                            marginBottom: 22,
                                            fontSize: 16,
                                            color: "#526282",
                                        } }, { children: "Analyses previous tax returns to identify missed deduction opportunities" })), _jsx(_Fragment, { children: loading ? (_jsx("div", Object.assign({ style: { padding: "10px 24px" } }, { children: _jsx("p", Object.assign({ style: {
                                                    fontSize: "0.875rem",
                                                    color: "#526282",
                                                    margin: "40px 0px",
                                                    display: "flex",
                                                    justifyContent: "center",
                                                    width: "100%",
                                                } }, { children: "Loading\u2026" })) }))) : (_jsx(_Fragment, { children: data ? (_jsx(TaxSummaryDashboard, { data: data })) : (_jsx("p", { children: "Loading tax data..." })) })) })] }))] })) })) })))] }));
};
export default TaxInsightWidget;
