// @rohit23061999/transaction-review-widget/dist/helper/AddTransactionForm.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useState } from "react";
import api from "../api/api";
import { toast } from "react-toastify";
const defaultFormData = {
    account_id: "",
    amount: "",
    iso_currency_code: "USD",
    datetime: new Date().toISOString(),
    date: new Date().toISOString(),
    name: "",
    merchant_name: "",
    payment_channel: "",
    transaction_id: "",
    transaction_type: "",
    personal_finance_category: {
        primary: "",
        detailed: "",
    },
};
const AddTransactionForm = ({ userId, accessToken, onSuccess, onError, }) => {
    const [formData, setFormData] = useState(defaultFormData);
    const [submitting, setSubmitting] = useState(false);
    const [error, setError] = useState("");
    const [fieldErrors, setFieldErrors] = useState({});
    const validate = () => {
        const amountNum = Number(formData.amount);
        const errors = {};
        if (!formData.name.trim())
            errors.name = "Transaction name is required.";
        if (!formData.merchant_name.trim())
            errors.merchant_name = "Merchant name is required.";
        if (!amountNum || Number(amountNum <= 0))
            errors.amount = "Amount must be greater than 0.";
        if (!formData.account_id.trim())
            errors.account_id = "Account ID is required.";
        if (!formData.transaction_id.trim())
            errors.transaction_id = "Transaction ID is required.";
        if (!formData.payment_channel.trim())
            errors.payment_channel = "Payment channel is required.";
        if (!formData.transaction_type.trim())
            errors.transaction_type = "Transaction type is required.";
        if (!formData.personal_finance_category.primary.trim())
            errors.primary = "Primary category is required.";
        if (!formData.personal_finance_category.detailed.trim())
            errors.detailed = "Detailed category is required.";
        setFieldErrors(errors);
        return Object.keys(errors).length === 0;
    };
    const clearError = (field) => {
        if (fieldErrors[field]) {
            setFieldErrors((prev) => (Object.assign(Object.assign({}, prev), { [field]: "" })));
        }
    };
    const handleSubmit = () => __awaiter(void 0, void 0, void 0, function* () {
        var _a, _b;
        if (!validate())
            return;
        setSubmitting(true);
        setError("");
        const payload = {
            user_id: userId,
            transactions: [formData],
        };
        try {
            const response = yield api.post("/v2/api/transactions", payload);
            const data = response.data;
            setFormData(defaultFormData);
            setFieldErrors({});
            onSuccess();
            toast.success("Transaction added successfully", { toastId: "Transaction added successfully" });
        }
        catch (err) {
            const msg = ((_b = (_a = err === null || err === void 0 ? void 0 : err.response) === null || _a === void 0 ? void 0 : _a.data) === null || _b === void 0 ? void 0 : _b.detail) || "Failed to add transaction";
            setError(msg);
            onError === null || onError === void 0 ? void 0 : onError(msg);
            toast.error(msg, { toastId: "error-msg" });
        }
        finally {
            setSubmitting(false);
        }
    });
    return (_jsxs("div", Object.assign({ style: {
            width: "100%",
            maxHeight: "70vh",
            backgroundColor: "#fff",
            borderRadius: "12px",
            boxShadow: "0 0 2px rgba(0,0,0,0.2)",
            margin: "auto",
            zIndex: 9998,
            padding: "0",
            overflowY: "auto",
        } }, { children: [_jsx("h2", Object.assign({ style: {
                    fontSize: "1.125rem",
                    fontWeight: "500",
                    color: "#042567",
                    margin: "0px",
                    backgroundColor: "#F1F7FB",
                    padding: "16px 24px",
                    display: "flex",
                    justifyContent: "space-between",
                    alignItems: "center",
                    borderBottom: "1px solid #BFDBFE",
                    marginBottom: "18px",
                } }, { children: "Add New Transaction" })), _jsxs("div", Object.assign({ style: { padding: "8px 16px" } }, { children: [error && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: error }))), _jsxs("div", Object.assign({ style: {
                            display: "flex",
                            alignItems: "start",
                            gap: "16px",
                            marginBottom: "12px",
                        } }, { children: [_jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Transaction Name" })), _jsx("input", { type: "text", placeholder: "Transaction Name", value: formData.name, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { name: e.target.value }));
                                                    clearError("name");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.name && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.name })))] })), _jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Merchant Name" })), _jsx("input", { placeholder: "Merchant Name", value: formData.merchant_name, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { merchant_name: e.target.value }));
                                                    clearError("merchant_name");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.merchant_name && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.merchant_name })))] }))] })), _jsxs("div", Object.assign({ style: {
                            display: "flex",
                            alignItems: "start",
                            gap: "16px",
                            marginBottom: "12px",
                        } }, { children: [_jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Amount" })), _jsx("input", { placeholder: "Amount", type: "text", value: formData.amount, onChange: (e) => {
                                                    const value = e.target.value;
                                                    // Allow empty input
                                                    if (value === "") {
                                                        setFormData(Object.assign(Object.assign({}, formData), { amount: "" }));
                                                        clearError("amount");
                                                        return;
                                                    }
                                                    // Regex: start with optional digits, optional single decimal, optional digits after decimal
                                                    // Prevent multiple decimals, letters, or invalid characters
                                                    const valid = /^(\d+(\.\d*)?|\.\d*)$/;
                                                    if (valid.test(value)) {
                                                        setFormData(Object.assign(Object.assign({}, formData), { amount: value }));
                                                        clearError("amount");
                                                    }
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.amount && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.amount })))] })), _jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Account ID" })), _jsx("input", { placeholder: "Account ID", value: formData.account_id, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { account_id: e.target.value }));
                                                    clearError("account_id");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.account_id && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.account_id })))] }))] })), _jsxs("div", Object.assign({ style: {
                            display: "flex",
                            alignItems: "start",
                            gap: "16px",
                            marginBottom: "12px",
                        } }, { children: [_jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Transaction ID" })), _jsx("input", { placeholder: "Transaction ID", value: formData.transaction_id, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { transaction_id: e.target.value }));
                                                    clearError("transaction_id");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.transaction_id && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.transaction_id })))] })), _jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Payment Channel" })), _jsx("input", { placeholder: "Payment Channel", value: formData.payment_channel, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { payment_channel: e.target.value }));
                                                    clearError("payment_channel");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.payment_channel && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.payment_channel })))] }))] })), _jsxs("div", Object.assign({ style: {
                            display: "flex",
                            alignItems: "start",
                            gap: "16px",
                            marginBottom: "12px",
                        } }, { children: [_jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Transaction Type" })), _jsx("input", { placeholder: "Transaction Type", value: formData.transaction_type, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { transaction_type: e.target.value }));
                                                    clearError("transaction_type");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.transaction_type && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.transaction_type })))] })), _jsxs("div", Object.assign({ style: { width: "49%" } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                                    fontSize: "14px",
                                                    fontWeight: "500",
                                                    color: "#042567",
                                                    marginBottom: "5px",
                                                    display: "block",
                                                } }, { children: "Primary Category" })), _jsx("input", { placeholder: "Primary Category", value: formData.personal_finance_category.primary, onChange: (e) => {
                                                    setFormData(Object.assign(Object.assign({}, formData), { personal_finance_category: Object.assign(Object.assign({}, formData.personal_finance_category), { primary: e.target.value }) }));
                                                    clearError("primary");
                                                }, style: {
                                                    width: "calc(100% - 35px)",
                                                    padding: "10px 16px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "10px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] })), fieldErrors.primary && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.primary })))] }))] })), _jsxs("div", Object.assign({ style: {
                            display: "block",
                            marginBottom: "0px",
                        } }, { children: [_jsxs("div", Object.assign({ style: { width: "100%" } }, { children: [_jsx("label", Object.assign({ style: {
                                            fontSize: "14px",
                                            fontWeight: "500",
                                            color: "#042567",
                                            marginBottom: "5px",
                                            display: "block",
                                        } }, { children: "Detailed Category" })), _jsx("input", { placeholder: "Detailed Category", value: formData.personal_finance_category.detailed, onChange: (e) => {
                                            setFormData(Object.assign(Object.assign({}, formData), { personal_finance_category: Object.assign(Object.assign({}, formData.personal_finance_category), { detailed: e.target.value }) }));
                                            clearError("detailed");
                                        }, style: {
                                            width: "calc(100% - 35px)",
                                            padding: "10px 16px",
                                            border: "1px solid #E6E7EA",
                                            borderRadius: "10px",
                                            fontSize: "0.875rem",
                                            color: "#042567",
                                            outline: "none",
                                        } })] })), fieldErrors.detailed && (_jsx("p", Object.assign({ style: { color: "red", fontSize: "12px", fontWeight: "400" } }, { children: fieldErrors.detailed })))] }))] })), _jsxs("div", Object.assign({ style: {
                    display: "flex",
                    alignItems: "center",
                    justifyContent: "center",
                    marginTop: "20px",
                    marginBottom: "20px",
                    gap: "10px",
                } }, { children: [_jsx("button", Object.assign({ disabled: submitting, onClick: onSuccess, style: {
                            border: "1px solid #5387F1",
                            backgroundColor: "white",
                            color: "#5387F1",
                            fontSize: "0.875rem",
                            fontWeight: 500,
                            padding: "8px 16px",
                            borderRadius: 8,
                            cursor: "pointer",
                        } }, { children: "Cancel" })), _jsx("button", Object.assign({ onClick: handleSubmit, disabled: submitting, style: {
                            backgroundColor: "#5387F1",
                            border: "1px solid #5387F1",
                            color: "white",
                            fontSize: "0.875rem",
                            fontWeight: 500,
                            padding: "8px 16px",
                            borderRadius: 8,
                            marginRight: 8,
                            cursor: "pointer",
                        } }, { children: submitting ? "Submitting..." : "Submit" }))] }))] })));
};
export default AddTransactionForm;
