// @rohit23061999/transaction-review-widget/dist/helper/Error.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
const ErrorWidget = ({ message, onClose }) => {
    console.log(message, 'messagemessagemessage');
    return (_jsxs("div", Object.assign({ style: {
            display: "flex",
            alignItems: "center",
            flexDirection: "column",
            justifyContent: "center",
            height: "100%",
        } }, { children: [_jsx("h2", Object.assign({ style: {
                    color: "#000000",
                    margin: "0px 0px 10px 0px",
                    fontSize: "22px",
                } }, { children: "Oops! Something went wrong" })), _jsx("p", Object.assign({ style: {
                    fontSize: "16px",
                    color: "#333333",
                    textAlign: "center",
                    margin: "0px",
                } }, { children: message })), _jsxs("button", Object.assign({ id: "close-widget-btn", onClick: () => onClose(), style: {
                    background: "#333333",
                    padding: "6px 16px",
                    color: "#ffffff",
                    borderRadius: "8px",
                    textAlign: "center",
                    fontSize: "16px",
                    margin: "20px 0px 0px 0px",
                    border: "none",
                    cursor: "pointer",
                    display: "flex",
                    alignItems: "center",
                    gap: "8px",
                } }, { children: [_jsx("i", { className: "fas fa-times" }), " Close"] }))] })));
};
export default ErrorWidget;
