// @rohit23061999/transaction-review-widget/dist/helper/TaxSummaryDashboard.js
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import { useState } from "react";
const TaxSummaryDashboard = ({ data }) => {
    var _a, _b, _c, _d, _e;
    const [activeSection, setActiveSection] = useState(null);
    const sectionStyle = {
        marginBottom: "30px",
        padding: "20px",
        backgroundColor: "#ffffff",
        borderRadius: "12px",
        boxShadow: "0 2px 8px rgba(0,0,0,0.1)",
    };
    const tableStyle = {
        width: "100%",
        borderCollapse: "collapse",
        marginTop: "0px",
    };
    const thTdStyle = {
        padding: "12px",
        borderBottom: "1px solid #ddd",
        textAlign: "left",
    };
    const sectionHeaderStyle = {
        display: "flex",
        justifyContent: "space-between",
        alignItems: "center",
        marginBottom: "10px",
    };
    const buttonStyle = (isOpen) => ({
        padding: "6px 14px",
        fontSize: "14px",
        backgroundColor: isOpen ? "#dc3545" : "#28a745",
        color: "#fff",
        border: "none",
        borderRadius: "6px",
        cursor: "pointer",
    });
    const sectionHeadingColor = (label) => {
        const colorMap = {
            "Schedule A - Deductions": "#6f42c1",
            "Schedule D - Investment": "#007bff",
            "Schedule E - Rentals": "#20c997",
            "Education Expenses": "#fd7e14",
        };
        return colorMap[label] || "#343a40";
    };
    const renderTransactions = (transactions) => (_jsx("div", Object.assign({ className: "custom-scrollbar", style: { height: "400px", overflowY: "auto", marginTop: "20px" } }, { children: _jsxs("table", Object.assign({ style: tableStyle }, { children: [_jsx("thead", { children: _jsxs("tr", { children: [_jsx("th", Object.assign({ style: thTdStyle }, { children: "Name" })), _jsx("th", Object.assign({ style: thTdStyle }, { children: "Amount" })), _jsx("th", Object.assign({ style: thTdStyle }, { children: "Date" })), _jsx("th", Object.assign({ style: thTdStyle }, { children: "Merchant" }))] }) }), _jsx("tbody", { children: transactions.map((tx, idx) => (_jsxs("tr", { children: [_jsx("td", Object.assign({ style: thTdStyle }, { children: tx.name })), _jsxs("td", Object.assign({ style: thTdStyle }, { children: ["$", tx.amount.toLocaleString()] })), _jsx("td", Object.assign({ style: thTdStyle }, { children: tx.date })), _jsx("td", Object.assign({ style: thTdStyle }, { children: tx.merchant }))] }, idx))) })] })) })));
    const renderScheduleSection = (label, item) => {
        const isOpen = activeSection === label;
        return (_jsxs(_Fragment, { children: [_jsx("style", { children: `
        .custom-scrollbar::-webkit-scrollbar {
          width: 8px;
        }

        .custom-scrollbar::-webkit-scrollbar-track {
          background: transparent;
          border-radius: 10px;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb {
          background-color: #ccc;
          border-radius: 10px;
          border: 2px solid transparent;
          background-clip: content-box;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb:hover {
          background-color: #999;
        }

        .custom-scrollbar {
          scrollbar-width: thin;
          scrollbar-color: #ccc transparent;
        }
      ` }), _jsxs("div", Object.assign({ style: sectionStyle }, { children: [_jsxs("div", Object.assign({ style: sectionHeaderStyle }, { children: [_jsxs("h2", Object.assign({ style: {
                                        color: sectionHeadingColor(label),
                                        margin: "0px",
                                        fontSize: "18px",
                                    } }, { children: [label, " - $", item.total.toLocaleString()] })), _jsx("button", Object.assign({ style: buttonStyle(isOpen), onClick: () => setActiveSection(isOpen ? null : label) }, { children: isOpen ? "Hide Details" : "Show Details" }))] })), _jsxs("p", { children: [_jsx("strong", { children: "Description:" }), " ", item.description] }), _jsxs("p", Object.assign({ style: { marginBottom: "0px" } }, { children: [_jsx("strong", { children: "Notes:" }), " ", item.notes] })), isOpen && renderTransactions(item.transactions)] }), label)] }));
    };
    const { schedule_a_deductions, schedule_d_items, schedule_e_items, education_expenses, summary, recommendations, } = data.analysis;
    const { form_requirements } = data;
    return (_jsxs("div", Object.assign({ style: {
            margin: "0 auto",
            padding: "0pxpx",
            fontFamily: "Arial",
        } }, { children: [_jsxs("div", Object.assign({ style: sectionStyle }, { children: [_jsx("h2", Object.assign({ style: { marginBottom: "16px", marginTop: "0px", fontSize: "20px" } }, { children: "Summary" })), _jsx("div", Object.assign({ style: {
                            display: "grid",
                            gridTemplateColumns: "1fr 1fr",
                            gap: "8px",
                        } }, { children: [
                            ["Schedule A Total", summary === null || summary === void 0 ? void 0 : summary.total_schedule_a],
                            ["Schedule D Total", summary === null || summary === void 0 ? void 0 : summary.total_schedule_d],
                            ["Schedule E Total", summary === null || summary === void 0 ? void 0 : summary.total_schedule_e],
                            ["Education Total", summary === null || summary === void 0 ? void 0 : summary.total_education],
                            ["AGI Adjustments", summary === null || summary === void 0 ? void 0 : summary.total_agi_adjustments],
                            ["Total Home Office", summary === null || summary === void 0 ? void 0 : summary.total_home_office],
                            ["Total HSA", summary === null || summary === void 0 ? void 0 : summary.total_hsa],
                            ["Total Retirement", summary === null || summary === void 0 ? void 0 : summary.total_retirement],
                            ["Total Itemized", (_a = summary === null || summary === void 0 ? void 0 : summary.itemized_vs_standard) === null || _a === void 0 ? void 0 : _a.itemized_total],
                            [
                                "Should Itemize MFJ",
                                (_b = summary === null || summary === void 0 ? void 0 : summary.itemized_vs_standard) === null || _b === void 0 ? void 0 : _b.should_itemize_mfj,
                            ],
                            [
                                "Should Itemize Single",
                                (_c = summary === null || summary === void 0 ? void 0 : summary.itemized_vs_standard) === null || _c === void 0 ? void 0 : _c.should_itemize_single,
                            ],
                            ["Standard MFJ", (_d = summary === null || summary === void 0 ? void 0 : summary.itemized_vs_standard) === null || _d === void 0 ? void 0 : _d.standard_mfj],
                            ["Standard Single", (_e = summary === null || summary === void 0 ? void 0 : summary.itemized_vs_standard) === null || _e === void 0 ? void 0 : _e.standard_single],
                            ["Standard Deduction (Single)", summary === null || summary === void 0 ? void 0 : summary.standard_deduction_2024],
                            ["Standard Deduction (MFJ)", summary === null || summary === void 0 ? void 0 : summary.standard_deduction_2024_mfj],
                        ].map(([label, value], idx) => {
                            var _a, _b;
                            return (_jsxs("div", Object.assign({ style: {
                                    backgroundColor: "#f8f9fa",
                                    padding: "12px 16px",
                                    borderRadius: "8px",
                                    display: "flex",
                                    justifyContent: "space-between",
                                    alignItems: "center",
                                    fontSize: "15px",
                                } }, { children: [_jsx("span", Object.assign({ style: { color: "#555" } }, { children: _jsx("strong", { children: label }) })), _jsxs("span", Object.assign({ style: { color: "#007bff", fontWeight: "bold" } }, { children: ["$", (_b = (_a = value === null || value === void 0 ? void 0 : value.toLocaleString) === null || _a === void 0 ? void 0 : _a.call(value)) !== null && _b !== void 0 ? _b : value] }))] }), idx));
                        }) }))] })), _jsx("h1", Object.assign({ style: { marginBottom: "16px", fontSize: "20px", fontWeight: "bold" } }, { children: "Tax Summary Dashboard" })), _jsx("h2", Object.assign({ style: {
                    marginBottom: "10px",
                    fontSize: "18px",
                    fontWeight: "bold",
                    color: sectionHeadingColor("Schedule A - Deductions"),
                } }, { children: "Schedule A - Deductions" })), Object.entries(schedule_a_deductions).map(([_, item]) => renderScheduleSection(item.description, item)), _jsx("h2", Object.assign({ style: {
                    marginBottom: "10px",
                    fontSize: "18px",
                    fontWeight: "bold",
                    color: sectionHeadingColor("Schedule D - Investment"),
                } }, { children: "Schedule D - Investment" })), Object.entries(schedule_d_items).map(([_, item]) => renderScheduleSection(item.description, item)), _jsx("h2", Object.assign({ style: {
                    marginBottom: "10px",
                    fontSize: "18px",
                    fontWeight: "bold",
                    color: sectionHeadingColor("Schedule E - Rentals"),
                } }, { children: "Schedule E - Rentals" })), Object.entries(schedule_e_items).map(([_, item]) => renderScheduleSection(item.description, item)), _jsx("h2", Object.assign({ style: {
                    marginBottom: "10px",
                    fontSize: "18px",
                    fontWeight: "bold",
                    color: sectionHeadingColor("Education Expenses"),
                } }, { children: "Education Expenses" })), Object.entries(education_expenses).map(([_, item]) => renderScheduleSection("Tuition & Fees", item)), _jsxs("div", Object.assign({ style: sectionStyle }, { children: [_jsx("h2", Object.assign({ style: { marginBottom: "16px", marginTop: "0px", fontSize: "20px" } }, { children: "Recommendations" })), _jsx("div", Object.assign({ style: { display: "grid", gap: "16px" } }, { children: recommendations.map((rec, idx) => (_jsxs("div", Object.assign({ style: {
                                border: "1px solid #eee",
                                borderRadius: "10px",
                                padding: "16px",
                                backgroundColor: "#fafafa",
                                boxShadow: "0 1px 4px rgba(0,0,0,0.05)",
                            } }, { children: [_jsxs("div", Object.assign({ style: {
                                        marginBottom: "8px",
                                        display: "flex",
                                        justifyContent: "space-between",
                                        alignItems: "center",
                                    } }, { children: [_jsx("strong", Object.assign({ style: { fontSize: "16px" } }, { children: rec.category })), _jsx("span", Object.assign({ style: {
                                                padding: "4px 10px",
                                                fontSize: "12px",
                                                borderRadius: "20px",
                                                backgroundColor: rec.priority === "High"
                                                    ? "#dc3545"
                                                    : rec.priority === "Medium"
                                                        ? "#ffc107"
                                                        : "#28a745",
                                                color: "#fff",
                                                fontWeight: "bold",
                                            } }, { children: rec.priority }))] })), _jsx("div", Object.assign({ style: { marginBottom: "10px", color: "#333" } }, { children: rec.recommendation })), rec.action && (_jsxs("div", Object.assign({ style: { marginBottom: "6px" } }, { children: [_jsx("strong", { children: "Action:" }), " ", _jsx("em", { children: rec.action })] }))), rec.potential_savings && (_jsxs("div", { children: [_jsx("strong", { children: "Potential Savings:" }), " ", _jsxs("span", Object.assign({ style: { color: "#28a745", fontWeight: "bold" } }, { children: ["$", rec.potential_savings.toFixed(2)] }))] }))] }), idx))) }))] })), _jsxs("div", Object.assign({ style: sectionStyle }, { children: [_jsx("h2", Object.assign({ style: { marginBottom: "16px", marginTop: "0px", fontSize: "20px" } }, { children: "Required Forms" })), _jsxs("table", Object.assign({ style: tableStyle }, { children: [_jsx("thead", { children: _jsxs("tr", { children: [_jsx("th", Object.assign({ style: thTdStyle }, { children: "Form" })), _jsx("th", Object.assign({ style: thTdStyle }, { children: "Required" })), _jsx("th", Object.assign({ style: thTdStyle }, { children: "Description" }))] }) }), _jsx("tbody", { children: Object.entries(form_requirements === null || form_requirements === void 0 ? void 0 : form_requirements.required_forms).map(([formKey, isRequired]) => (_jsxs("tr", { children: [_jsx("td", Object.assign({ style: thTdStyle }, { children: formKey.toUpperCase() })), _jsx("td", Object.assign({ style: thTdStyle }, { children: isRequired ? "✅ Yes" : "❌ No" })), _jsx("td", Object.assign({ style: thTdStyle }, { children: form_requirements.form_descriptions[formKey] }))] }, formKey))) })] }))] }))] })));
};
export default TaxSummaryDashboard;
