// @rohit23061999/transaction-review-widget/dist/transactionReviewWidget/ClientAuthForm.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import React, { useEffect, useMemo, useState } from "react";
import AddTransactionForm from "../helper/AddTransactionForm";
import api from "../api/api";
import ErrorWidget from "../helper/Error";
import { getAuthSessionToken } from "../helper/AuthHelper";
import { toast } from "react-toastify";
import { formatNumberWithCommas } from "../helper/NumberFormat";
const { CiSearch } = require("react-icons/ci");
const { FaPlus, FaCaretDown } = require("react-icons/fa6");
const { HiOutlineArrowSmLeft } = require("react-icons/hi");
const ClientAuthForm = ({ userId, access_token, session_token, onError, }) => {
    var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u, _v, _w;
    const [year, setYear] = useState("");
    const [isOpen, setIsOpen] = useState(false);
    const [selected, setSelected] = useState(null);
    const [search, setSearch] = useState("");
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState("");
    const [transactions, setTransactions] = useState([]);
    const [showAddForm, setShowAddForm] = useState(false);
    const [categoryNames, setCategoryNames] = useState([]);
    // done
    const fetchTransactions = (userId) => __awaiter(void 0, void 0, void 0, function* () {
        var _x, _y, _z, _0, _1;
        setLoading(true);
        try {
            const { data } = yield api.get(`/v2/api/deduction-dashboard/?user_id=${userId}&year=${year}`);
            console.log(data, "transactions");
            if (data === null || data === void 0 ? void 0 : data.detail) {
                // setError(data?.detail);
                onError === null || onError === void 0 ? void 0 : onError(data.detail);
                toast.error(data.detail, {
                    toastId: " data.detail-error-msg-error-Msg",
                });
                setLoading(false);
            }
            else {
                setTransactions((data === null || data === void 0 ? void 0 : data.data) || []);
                setLoading(false);
            }
        }
        catch (error) {
            console.log(error, "errorerrorerrorerror", (_y = (_x = error === null || error === void 0 ? void 0 : error.response) === null || _x === void 0 ? void 0 : _x.data) === null || _y === void 0 ? void 0 : _y.detail);
            console.log((_z = error === null || error === void 0 ? void 0 : error.response) === null || _z === void 0 ? void 0 : _z.data, "error?.response?.data?.detail ");
            const msg = ((_1 = (_0 = error === null || error === void 0 ? void 0 : error.response) === null || _0 === void 0 ? void 0 : _0.data) === null || _1 === void 0 ? void 0 : _1.detail) || "Failed to fetch transactions.";
            // setError(msg);
            onError === null || onError === void 0 ? void 0 : onError(msg);
            toast.error(msg, { toastId: "-msg-error-Msg" });
            setLoading(false);
        }
    });
    //done
    const getAllUserCategory = () => __awaiter(void 0, void 0, void 0, function* () {
        var _2, _3;
        try {
            const response = yield api.get(`/v2/api/categorized/data/${userId}`);
            const data = response.data;
            setCategoryNames(data); // 🔥 Save to state
        }
        catch (err) {
            console.error("Error fetching categories:", err);
            const msg = ((_3 = (_2 = err === null || err === void 0 ? void 0 : err.response) === null || _2 === void 0 ? void 0 : _2.data) === null || _3 === void 0 ? void 0 : _3.detail) || "Error fetching categories";
            toast.error(msg, { toastId: "e-msg-error-Msg" });
        }
    });
    // done
    const updateTransactionCategory = (payload) => __awaiter(void 0, void 0, void 0, function* () {
        try {
            console.log(payload, "payloadpayloadpayloadpayload");
            const response = yield api.put("/v2/api/transaction/update/category", payload);
            console.log(response, "responseresponseresponse");
            if ((response === null || response === void 0 ? void 0 : response.status) == 200) {
                setSelected(null);
                toast.success("Category updated successfully", {
                    toastId: "Category updated successfully",
                });
                yield fetchTransactions(userId);
            }
            else {
                toast.error("Something went wrong", {
                    toastId: "Something went wrong",
                });
            }
        }
        catch (err) {
            toast.error("Failed to update category", {
                toastId: "Failed to update category",
            });
        }
    });
    const allSubcategories = useMemo(() => {
        const result = {};
        Object.entries((transactions === null || transactions === void 0 ? void 0 : transactions.categories) || {}).forEach(([catKey, catValue]) => {
            const subs = (catValue === null || catValue === void 0 ? void 0 : catValue.subcategories) || {};
            Object.entries(subs).forEach(([subKey, subVal]) => {
                result[subKey] = Object.assign(Object.assign({}, subVal), { parentCategoryKey: catKey, parentCategoryName: catValue === null || catValue === void 0 ? void 0 : catValue.name, parentCategoryForm: catValue === null || catValue === void 0 ? void 0 : catValue.form, parentCategoryDescription: catValue === null || catValue === void 0 ? void 0 : catValue.description, parentCategoryEligible: catValue === null || catValue === void 0 ? void 0 : catValue.eligible, parentCategoryFoundCount: catValue === null || catValue === void 0 ? void 0 : catValue.found_count });
            });
        });
        return result;
    }, [transactions === null || transactions === void 0 ? void 0 : transactions.categories]);
    const filteredSubcategories = useMemo(() => {
        if (!search.trim())
            return Object.entries(allSubcategories);
        const searchTerm = search.toLowerCase();
        return Object.entries(allSubcategories).filter(([key, value]) => {
            var _a, _b, _c, _d, _e, _f, _g, _h;
            const name = ((_b = (_a = value === null || value === void 0 ? void 0 : value.name) === null || _a === void 0 ? void 0 : _a.toLowerCase) === null || _b === void 0 ? void 0 : _b.call(_a)) || "";
            const line = ((_d = (_c = value === null || value === void 0 ? void 0 : value.line) === null || _c === void 0 ? void 0 : _c.toLowerCase) === null || _d === void 0 ? void 0 : _d.call(_c)) || "";
            const parentName = ((_f = (_e = value === null || value === void 0 ? void 0 : value.parentCategoryName) === null || _e === void 0 ? void 0 : _e.toLowerCase) === null || _f === void 0 ? void 0 : _f.call(_e)) || "";
            const parentForm = ((_h = (_g = value === null || value === void 0 ? void 0 : value.parentCategoryForm) === null || _g === void 0 ? void 0 : _g.toLowerCase) === null || _h === void 0 ? void 0 : _h.call(_g)) || "";
            return (key.includes(searchTerm) ||
                name.includes(searchTerm) ||
                line.includes(searchTerm) ||
                parentName.includes(searchTerm) ||
                parentForm.includes(searchTerm));
        });
    }, [search, allSubcategories]);
    const taxSavings = 0;
    useEffect(() => {
        if (access_token && session_token) {
            getAuthSessionToken({
                session_token,
                userId,
                onSuccess: (data) => {
                    console.log("Auth token fetched, domain URLs:", data.domain_urls);
                    getAllUserCategory();
                },
                onError,
                setLoading,
                setError,
            });
        }
    }, [access_token, session_token]);
    useEffect(() => {
        if (year) {
            fetchTransactions(userId);
        }
    }, [year]);
    return (_jsxs(_Fragment, { children: [_jsx("style", { children: `
        .custom-scrollbar::-webkit-scrollbar {
          width: 8px;
        }

        .custom-scrollbar::-webkit-scrollbar-track {
          background: transparent;
          border-radius: 10px;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb {
          background-color: #ccc;
          border-radius: 10px;
          border: 2px solid transparent;
          background-clip: content-box;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb:hover {
          background-color: #999;
        }

        .custom-scrollbar {
          scrollbar-width: thin;
          scrollbar-color: #ccc transparent;
        }
      ` }), _jsx("div", Object.assign({ style: {
                    position: "fixed",
                    bottom: "20px",
                    right: "20px",
                    width: "60px",
                    height: "60px",
                    backgroundColor: "#007bff",
                    borderRadius: "50%",
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                    cursor: "pointer",
                    zIndex: 9999,
                    color: "#fff",
                    fontSize: "28px",
                    boxShadow: "0 4px 5px rgba(0, 0, 0, 0.3)",
                }, onClick: () => setIsOpen(!isOpen), title: "Open Auth Widget" }, { children: "\uD83D\uDCAC" })), isOpen && (_jsx("div", Object.assign({ className: "custom-scrollbar", style: {
                    position: "fixed",
                    bottom: "90px",
                    right: "20px",
                    width: "50rem",
                    maxHeight: "75vh",
                    backgroundColor: "#fff",
                    borderRadius: "12px",
                    boxShadow: "0 0 5px rgba(0,0,0,0.2)",
                    zIndex: 9998,
                    padding: "0",
                    overflowY: "auto",
                } }, { children: error ? (_jsx("div", Object.assign({ style: {
                        padding: "0",
                        minHeight: "500px",
                        paddingTop: "200px",
                    } }, { children: _jsx(ErrorWidget, { message: error, onClose: () => setIsOpen(false) }) }))) : (_jsxs("div", Object.assign({ style: {
                        padding: "0",
                        minHeight: "500px",
                    } }, { children: [_jsx("div", Object.assign({ style: {
                                backgroundColor: "#F1F7FB",
                                padding: "16px 24px",
                                display: "flex",
                                justifyContent: "space-between",
                                alignItems: "center",
                                borderBottom: "1px solid #BFDBFE",
                                marginBottom: "0px",
                            } }, { children: _jsx("h2", Object.assign({ style: {
                                    fontSize: "1.125rem",
                                    fontWeight: "500",
                                    color: "#042567",
                                    margin: "0px",
                                } }, { children: "Review & Categorize" })) })), !selected && (_jsx("div", Object.assign({ style: {
                                display: "flex",
                                alignItems: "center",
                                justifyContent: "end",
                                padding: "0px 10px",
                                marginBottom: "10px",
                                marginTop: "18px",
                            } }, { children: _jsxs("select", Object.assign({ id: "yearSelect", value: year || "", onChange: (e) => setYear(parseInt(e.target.value)), style: {
                                    padding: "8px",
                                    borderRadius: "6px",
                                    border: "1px solid rgba(30, 64, 175, 0.1)",
                                    width: "100%",
                                    outline: "none",
                                    background: "#ffffff",
                                } }, { children: [_jsx("option", Object.assign({ value: "", disabled: true }, { children: "Select Year" })), (() => {
                                        const currentYear = new Date().getFullYear();
                                        const years = Array.from({ length: currentYear }, (_, i) => currentYear - i);
                                        return years.map((y) => (_jsx("option", Object.assign({ value: y }, { children: y }), y)));
                                    })()] })) }))), _jsx(_Fragment, { children: year && (_jsxs(_Fragment, { children: [!selected && (_jsxs("div", Object.assign({ style: {
                                            padding: "0px 24px 16px",
                                            position: "relative",
                                        } }, { children: [_jsx("span", Object.assign({ style: {
                                                    position: "absolute",
                                                    top: "9px",
                                                    left: "40px",
                                                } }, { children: _jsx(CiSearch, { style: { color: "#526282", fontSize: "20px" } }) })), _jsx("input", { type: "text", value: search, onChange: (e) => setSearch(e.target.value), placeholder: "Search\u2026", style: {
                                                    width: "calc(100% - 70px)",
                                                    padding: "10px 16px 10px 44px",
                                                    border: "1px solid #E6E7EA",
                                                    borderRadius: "9999px",
                                                    fontSize: "0.875rem",
                                                    color: "#042567",
                                                    outline: "none",
                                                } })] }))), _jsxs("div", Object.assign({ style: { padding: "0 24px 10px" } }, { children: [!loading && (_jsx("div", Object.assign({ style: {
                                                    display: "flex",
                                                    alignItems: "center",
                                                    gap: "10px",
                                                    marginBottom: "16px",
                                                    justifyContent: "center",
                                                    flexWrap: "wrap",
                                                } }, { children: (filteredSubcategories === null || filteredSubcategories === void 0 ? void 0 : filteredSubcategories.length) == 0 ? ("") : (_jsx(_Fragment, { children: !selected && (_jsxs(_Fragment, { children: [_jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of completion percentage :", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_c = (_b = (_a = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _a === void 0 ? void 0 : _a.completion_percentage) === null || _b === void 0 ? void 0 : _b.toFixed(2)) !== null && _c !== void 0 ? _c : 0) }))] })), _jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of deductible transactions :", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_f = (_e = (_d = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _d === void 0 ? void 0 : _d.deductible_transactions) === null || _e === void 0 ? void 0 : _e.toFixed(2)) !== null && _f !== void 0 ? _f : 0) }))] })), _jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of potential savings :", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_j = (_h = (_g = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _g === void 0 ? void 0 : _g.potential_savings) === null || _h === void 0 ? void 0 : _h.toFixed(2)) !== null && _j !== void 0 ? _j : 0) }))] })), _jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of total deductible amount :", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_m = (_l = (_k = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _k === void 0 ? void 0 : _k.total_deductible_amount) === null || _l === void 0 ? void 0 : _l.toFixed(2)) !== null && _m !== void 0 ? _m : 0) }))] })), _jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of total deductions available :", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_p = (_o = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _o === void 0 ? void 0 : _o.total_deductions_available) !== null && _p !== void 0 ? _p : 0) }))] })), _jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of total deductions found :", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_r = (_q = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _q === void 0 ? void 0 : _q.total_deductions_found) !== null && _r !== void 0 ? _r : 0) }))] })), _jsxs("span", Object.assign({ style: {
                                                                    fontSize: "14px",
                                                                    color: "rgb(82, 98, 130)",
                                                                    margin: "0px",
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    gap: 6,
                                                                } }, { children: ["Number of transaction", " ", _jsx("span", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "16px",
                                                                            fontWeight: 500,
                                                                        } }, { children: formatNumberWithCommas((_s = transactions === null || transactions === void 0 ? void 0 : transactions.total_transactions) !== null && _s !== void 0 ? _s : 0) }))] }))] })) })) }))), showAddForm ? (_jsx(AddTransactionForm, { userId: userId, accessToken: access_token, onSuccess: () => __awaiter(void 0, void 0, void 0, function* () {
                                                    // await handleAddTransaction();
                                                    setShowAddForm(false);
                                                    yield fetchTransactions(userId);
                                                }), onError: onError })) : loading ? (_jsx("p", Object.assign({ style: {
                                                    fontSize: "0.875rem",
                                                    color: "#526282",
                                                    margin: 0,
                                                    display: "flex",
                                                    justifyContent: "center",
                                                    width: "100%",
                                                } }, { children: "Loading\u2026" }))) : !selected ? (filteredSubcategories.length === 0 ? (_jsx("p", Object.assign({ style: {
                                                    fontSize: "0.875rem",
                                                    color: "#526282",
                                                    margin: 0,
                                                    display: "flex",
                                                    justifyContent: "center",
                                                    width: "100%",
                                                } }, { children: "No transactions match." }))) : (filteredSubcategories.map(([key, g]) => {
                                                var _a, _b;
                                                return (_jsxs("div", Object.assign({ onClick: () => setSelected(g), style: {
                                                        cursor: "pointer",
                                                        display: "flex",
                                                        justifyContent: "space-between",
                                                        alignItems: "center",
                                                        padding: "16px 20px",
                                                        border: "1px solid #1E40AF1A",
                                                        borderRadius: 12,
                                                        marginBottom: "12px",
                                                    } }, { children: [_jsxs("div", { children: [_jsx("h3", Object.assign({ style: {
                                                                        fontSize: "15px",
                                                                        fontWeight: 500,
                                                                        color: "#042567",
                                                                        margin: "0 0 6px 0",
                                                                    } }, { children: key !== null && key !== void 0 ? key : "NA" })), _jsx("span", Object.assign({ style: {
                                                                        fontSize: "0.75rem",
                                                                        backgroundColor: "#4A6CF71F",
                                                                        color: "#4A6CF7",
                                                                        borderRadius: 9999,
                                                                        padding: "4px 16px",
                                                                        display: "inline-block",
                                                                        border: "1px solid #4A6CF71A",
                                                                    } }, { children: ((_a = g === null || g === void 0 ? void 0 : g.parentCategoryName) !== null && _a !== void 0 ? _a : "") !== ""
                                                                        ? g === null || g === void 0 ? void 0 : g.parentCategoryName
                                                                        : "N/A" }))] }), _jsxs("div", Object.assign({ style: { textAlign: "right" } }, { children: [_jsxs("p", Object.assign({ style: {
                                                                        color: "#042567",
                                                                        fontSize: "16px",
                                                                        fontWeight: 500,
                                                                        margin: "0 0 4px",
                                                                        display: "flex",
                                                                        alignItems: "center",
                                                                        gap: 6,
                                                                    } }, { children: ["$", formatNumberWithCommas((_b = g.total_amount.toFixed(2)) !== null && _b !== void 0 ? _b : 0), " ", _jsx(FaCaretDown, {})] })), _jsxs("p", Object.assign({ style: {
                                                                        fontSize: "0.875rem",
                                                                        color: "#526282",
                                                                        margin: 0,
                                                                    } }, { children: [" ", g.transactions.length, " transactions"] }))] }))] }), g.category));
                                            }))) : (_jsx(_Fragment, { children: _jsxs("div", Object.assign({ style: {
                                                        backgroundColor: "#fff",
                                                        color: "#000",
                                                        borderRadius: "12px",
                                                        padding: "0px",
                                                        maxWidth: "100%",
                                                        margin: "0px",
                                                        fontFamily: "sans-serif",
                                                        fontSize: "0.9rem",
                                                    } }, { children: [_jsxs("button", Object.assign({ onClick: () => {
                                                                setSelected(null);
                                                                setSearch("");
                                                            }, style: {
                                                                background: "transparent",
                                                                border: "none",
                                                                color: "#2094EA",
                                                                marginBottom: "1rem",
                                                                cursor: "pointer",
                                                                fontWeight: 500,
                                                                fontSize: "1rem",
                                                                display: "flex",
                                                                alignItems: "center",
                                                                gap: "5px",
                                                            } }, { children: [_jsx(HiOutlineArrowSmLeft, { style: { fontSize: "22px" } }), "Back"] })), _jsxs("div", Object.assign({ style: {
                                                                padding: "0px",
                                                            } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                        backgroundColor: "#F1F7FB",
                                                                        borderRadius: "8px",
                                                                        padding: "1rem",
                                                                        display: "flex",
                                                                        justifyContent: "space-between",
                                                                        alignItems: "center",
                                                                        marginBottom: "14px",
                                                                        flexWrap: "wrap",
                                                                    } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                                display: "flex",
                                                                                flexDirection: "column",
                                                                            } }, { children: [_jsx("h2", Object.assign({ style: {
                                                                                        color: "#042567",
                                                                                        fontSize: "16px",
                                                                                        fontWeight: "500",
                                                                                        margin: 0,
                                                                                        textTransform: "capitalize",
                                                                                    } }, { children: selected === null || selected === void 0 ? void 0 : selected.name })), _jsxs("span", Object.assign({ style: {
                                                                                        backgroundColor: "#67DC791F",
                                                                                        color: "#15803D",
                                                                                        padding: "6px 16px",
                                                                                        border: "1px solid #4EC7601A",
                                                                                        borderRadius: "50px",
                                                                                        fontSize: "0.85rem",
                                                                                        marginTop: "6px",
                                                                                        textAlign: "center",
                                                                                    } }, { children: ["Deduction Opportunity", " "] }))] })), _jsxs("div", Object.assign({ style: { textAlign: "right" } }, { children: [_jsxs("p", Object.assign({ style: {
                                                                                        color: "#042567",
                                                                                        fontSize: "17px",
                                                                                        fontWeight: "500",
                                                                                        margin: 0,
                                                                                    } }, { children: ["$", formatNumberWithCommas((_t = selected.total_amount.toFixed(2)) !== null && _t !== void 0 ? _t : 0)] })), _jsxs("span", Object.assign({ style: {
                                                                                        fontSize: "14px",
                                                                                        color: "#526282",
                                                                                    } }, { children: [selected.transactions.length, " transactions"] }))] }))] })), _jsx("div", Object.assign({ style: {
                                                                        border: "1px solid #E4EEF4",
                                                                        borderRadius: "12px",
                                                                        overflow: "hidden",
                                                                    } }, { children: _jsxs("table", Object.assign({ style: {
                                                                            width: "100%",
                                                                            borderCollapse: "collapse",
                                                                            background: "#fff",
                                                                            borderRadius: "8px",
                                                                            overflow: "hidden",
                                                                        } }, { children: [_jsx("thead", Object.assign({ style: {
                                                                                    backgroundColor: "#f1f5f9",
                                                                                    textAlign: "left",
                                                                                } }, { children: _jsxs("tr", { children: [_jsx("th", Object.assign({ style: {
                                                                                                padding: "12px 16px",
                                                                                                color: "#042567",
                                                                                                textAlign: "left",
                                                                                                fontWeight: "500",
                                                                                            } }, { children: "Description" })), _jsx("th", Object.assign({ style: {
                                                                                                padding: "12px 16px",
                                                                                                color: "#042567",
                                                                                                textAlign: "left",
                                                                                                fontWeight: "500",
                                                                                            } }, { children: "Type" })), _jsx("th", Object.assign({ style: {
                                                                                                padding: "12px 16px",
                                                                                                color: "#042567",
                                                                                                textAlign: "left",
                                                                                                fontWeight: "500",
                                                                                            } }, { children: "Amount" })), _jsx("th", Object.assign({ style: {
                                                                                                padding: "12px 16px",
                                                                                                color: "#042567",
                                                                                                textAlign: "left",
                                                                                                fontWeight: "500",
                                                                                            } }, { children: "Confidence Score" })), _jsx("th", Object.assign({ style: {
                                                                                                padding: "12px 16px",
                                                                                                color: "#042567",
                                                                                                textAlign: "left",
                                                                                                fontWeight: "500",
                                                                                            } }, { children: "Date" })), _jsx("th", Object.assign({ style: {
                                                                                                padding: "12px 16px",
                                                                                                color: "#042567",
                                                                                                textAlign: "left",
                                                                                                fontWeight: "500",
                                                                                            } }, { children: "Category" }))] }) })), _jsx("tbody", { children: (_u = selected === null || selected === void 0 ? void 0 : selected.transactions) === null || _u === void 0 ? void 0 : _u.map((t) => {
                                                                                    var _a, _b, _c, _d, _e, _f;
                                                                                    const transactionId = (t === null || t === void 0 ? void 0 : t.id) || (t === null || t === void 0 ? void 0 : t.description);
                                                                                    // Deduplicate category names by category_details.name
                                                                                    const uniqueCategories = Array.from(new Map(categoryNames.map((item) => [
                                                                                        item === null || item === void 0 ? void 0 : item.category,
                                                                                        item,
                                                                                    ])).values());
                                                                                    return (_jsx(React.Fragment, { children: _jsxs("tr", Object.assign({ style: {
                                                                                                borderBottom: "1px solid #e0e0e0",
                                                                                            } }, { children: [_jsx("td", Object.assign({ style: {
                                                                                                        padding: "12px",
                                                                                                        color: "#444",
                                                                                                        textAlign: "left",
                                                                                                    } }, { children: (_a = t.name) !== null && _a !== void 0 ? _a : t.merchant_name })), _jsx("td", Object.assign({ style: {
                                                                                                        padding: "12px",
                                                                                                        textAlign: "left",
                                                                                                    } }, { children: ((_b = t.schedule) !== null && _b !== void 0 ? _b : "") !== ""
                                                                                                        ? t.schedule
                                                                                                        : ((_c = selected === null || selected === void 0 ? void 0 : selected.schedule) !== null && _c !== void 0 ? _c : "") !==
                                                                                                            ""
                                                                                                            ? selected.schedule
                                                                                                            : "N/A" })), _jsx("td", Object.assign({ style: {
                                                                                                        padding: "12px",
                                                                                                        textAlign: "left",
                                                                                                        color: t.amount < 0
                                                                                                            ? "#d32f2f"
                                                                                                            : "#2e7d32",
                                                                                                    } }, { children: formatNumberWithCommas((_e = (_d = t === null || t === void 0 ? void 0 : t.amount) === null || _d === void 0 ? void 0 : _d.toFixed(2)) !== null && _e !== void 0 ? _e : 0) })), _jsx("td", Object.assign({ style: {
                                                                                                        padding: "12px",
                                                                                                        textAlign: "left",
                                                                                                    } }, { children: (_f = t === null || t === void 0 ? void 0 : t.confidence_score) !== null && _f !== void 0 ? _f : 0 })), _jsx("td", Object.assign({ style: {
                                                                                                        padding: "12px",
                                                                                                        textAlign: "left",
                                                                                                    } }, { children: (t === null || t === void 0 ? void 0 : t.date)
                                                                                                        ? new Date(t.date).toLocaleDateString("en-US", {
                                                                                                            year: "numeric",
                                                                                                            month: "short",
                                                                                                            day: "numeric",
                                                                                                        })
                                                                                                        : "N/A" })), _jsx("td", Object.assign({ style: {
                                                                                                        padding: "12px",
                                                                                                        textAlign: "left",
                                                                                                    } }, { children: _jsxs("select", Object.assign({ value: (t === null || t === void 0 ? void 0 : t.category) || "", onChange: (e) => {
                                                                                                            var _a, _b, _c, _d;
                                                                                                            const selectedCategory = uniqueCategories.find((c) => (c === null || c === void 0 ? void 0 : c.category) ===
                                                                                                                e.target.value);
                                                                                                            if (selectedCategory) {
                                                                                                                const payload = {
                                                                                                                    transaction_id: t === null || t === void 0 ? void 0 : t.id,
                                                                                                                    category: selectedCategory === null || selectedCategory === void 0 ? void 0 : selectedCategory.category,
                                                                                                                    category_name: selectedCategory === null || selectedCategory === void 0 ? void 0 : selectedCategory.category,
                                                                                                                    schedule: (_a = selectedCategory === null || selectedCategory === void 0 ? void 0 : selectedCategory.category_details) === null || _a === void 0 ? void 0 : _a.schedule,
                                                                                                                    line: (_b = selectedCategory === null || selectedCategory === void 0 ? void 0 : selectedCategory.category_details) === null || _b === void 0 ? void 0 : _b.line,
                                                                                                                    deductible: (_c = selectedCategory === null || selectedCategory === void 0 ? void 0 : selectedCategory.category_details) === null || _c === void 0 ? void 0 : _c.deductible,
                                                                                                                    taxable: (_d = selectedCategory === null || selectedCategory === void 0 ? void 0 : selectedCategory.category_details) === null || _d === void 0 ? void 0 : _d.taxable,
                                                                                                                    confidence_score: 1,
                                                                                                                };
                                                                                                                updateTransactionCategory(payload);
                                                                                                            }
                                                                                                        }, style: {
                                                                                                            padding: "8px 16px",
                                                                                                            border: "1px solid #E6E7EA",
                                                                                                            borderRadius: "9999px",
                                                                                                            fontSize: "0.875rem",
                                                                                                            color: "#042567",
                                                                                                            outline: "none",
                                                                                                        } }, { children: [_jsx("option", Object.assign({ value: "" }, { children: "Select Category" })), uniqueCategories.map((data, index) => (_jsx("option", Object.assign({ value: data === null || data === void 0 ? void 0 : data.category }, { children: data === null || data === void 0 ? void 0 : data.category }), index)))] })) }))] })) }, transactionId));
                                                                                }) })] })) }))] }))] })) }))] })), !selected &&
                                        filteredSubcategories.length > 0 &&
                                        !showAddForm &&
                                        !loading && (_jsx("div", Object.assign({ style: { padding: "0 24px 24px" } }, { children: _jsxs("div", Object.assign({ style: {
                                                backgroundColor: "#F1F7FB",
                                                padding: "32px 24px",
                                                borderRadius: 16,
                                                textAlign: "center",
                                            } }, { children: [_jsxs("p", Object.assign({ style: {
                                                        fontSize: "14px",
                                                        fontWeight: 500,
                                                        color: "#042567",
                                                        marginBottom: 16,
                                                        marginTop: 0,
                                                    } }, { children: ["Potential Tax Savings from Reviewed Transactions: \u00A0", _jsxs("span", Object.assign({ style: { color: "#042567", fontWeight: "bold" } }, { children: ["$", " ", formatNumberWithCommas((_w = (_v = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _v === void 0 ? void 0 : _v.potential_savings) !== null && _w !== void 0 ? _w : 0)] }))] })), _jsx("button", Object.assign({ style: {
                                                        backgroundColor: "#5387F1",
                                                        border: "1px solid #5387F1",
                                                        color: "white",
                                                        fontSize: "0.875rem",
                                                        fontWeight: 500,
                                                        padding: "8px 16px",
                                                        borderRadius: 8,
                                                        marginRight: 8,
                                                        cursor: "pointer",
                                                    } }, { children: "Proceed" })), _jsx("button", Object.assign({ style: {
                                                        border: "1px solid #5387F1",
                                                        backgroundColor: "white",
                                                        color: "#5387F1",
                                                        fontSize: "0.875rem",
                                                        fontWeight: 500,
                                                        padding: "8px 16px",
                                                        borderRadius: 8,
                                                        cursor: "pointer",
                                                    } }, { children: "Review More" }))] })) })))] })) })] }))) })))] }));
};
export default ClientAuthForm;
