// @rohit23061999/transaction-review-widget/dist/potentialDeductionsWidget/PotentialDeductions.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import React, { useEffect, useMemo, useState } from "react";
import api from "../api/api";
import { getAuthSessionToken } from "../helper/AuthHelper";
import ErrorWidget from "../helper/Error";
import { toast } from "react-toastify";
import { formatNumberWithCommas } from "../helper/NumberFormat";
const { FaPlus, FaCaretDown } = require("react-icons/fa6");
const { HiOutlineArrowSmLeft } = require("react-icons/hi");
const PotentialDeductions = ({ userId, access_token, session_token, onError, }) => {
    var _a, _b, _c, _d, _e, _f, _g, _h;
    const [year, setYear] = useState("");
    const [transactions, setTransactions] = useState([]);
    const [isOpen, setIsOpen] = useState(false);
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState("");
    const [selected, setSelected] = useState(null);
    const fetchTransactions = (userId) => __awaiter(void 0, void 0, void 0, function* () {
        setLoading(true);
        try {
            const { data } = yield api.get(`/v2/api/deduction-dashboard/?user_id=${userId}&year=${year}`);
            console.log(data, "datadatadatadatadatadata");
            if (data === null || data === void 0 ? void 0 : data.detail) {
                const msg = data.detail;
                // setError(msg);
                onError === null || onError === void 0 ? void 0 : onError(msg);
                toast.error(msg, { toastId: "msg-error-msg-errorMsg" });
            }
            else {
                setTransactions((data === null || data === void 0 ? void 0 : data.data) || []);
            }
        }
        catch (error) {
            const msg = "Failed to fetch transactions.";
            // setError(msg);
            onError === null || onError === void 0 ? void 0 : onError(msg);
            toast.error(msg, { toastId: "msg-error-msg-error-Msg" });
        }
        finally {
            setLoading(false);
        }
    });
    useEffect(() => {
        if (access_token && session_token) {
            getAuthSessionToken({
                session_token,
                userId,
                onSuccess: () => {
                    // fetchTransactions(userId);
                },
                onError,
                setLoading,
                setError,
            });
        }
    }, [access_token, session_token]);
    const allSubcategories = useMemo(() => {
        const result = {};
        Object.entries((transactions === null || transactions === void 0 ? void 0 : transactions.categories) || {}).forEach(([catKey, catValue]) => {
            const subs = (catValue === null || catValue === void 0 ? void 0 : catValue.subcategories) || {};
            Object.entries(subs).forEach(([subKey, subVal]) => {
                result[subKey] = Object.assign(Object.assign({}, subVal), { parentCategoryKey: catKey, parentCategoryName: catValue === null || catValue === void 0 ? void 0 : catValue.name, parentCategoryForm: catValue === null || catValue === void 0 ? void 0 : catValue.form, parentCategoryDescription: catValue === null || catValue === void 0 ? void 0 : catValue.description, parentCategoryEligible: catValue === null || catValue === void 0 ? void 0 : catValue.eligible, parentCategoryFoundCount: catValue === null || catValue === void 0 ? void 0 : catValue.found_count });
            });
        });
        return result;
    }, [transactions === null || transactions === void 0 ? void 0 : transactions.categories]);
    const filteredSubcategories = useMemo(() => {
        return Object.entries(allSubcategories);
    }, [allSubcategories]);
    useEffect(() => {
        if (year) {
            fetchTransactions("6875de30cff6c49154070792");
        }
    }, [year]);
    const [selectedCategoryKey, setSelectedCategoryKey] = useState(null);
    const filtered = selectedCategoryKey
        ? filteredSubcategories.filter(([_, g]) => (g === null || g === void 0 ? void 0 : g.parentCategoryKey) === selectedCategoryKey)
        : filteredSubcategories;
    useEffect(() => {
        if (!selectedCategoryKey && (transactions === null || transactions === void 0 ? void 0 : transactions.categories)) {
            const firstKey = Object.keys(transactions.categories)[0];
            if (firstKey) {
                setSelectedCategoryKey(firstKey);
            }
        }
    }, [transactions === null || transactions === void 0 ? void 0 : transactions.categories]);
    return (_jsxs(_Fragment, { children: [_jsx("style", { children: `
        .custom-scrollbar::-webkit-scrollbar {
          width: 8px;
        }

        .custom-scrollbar::-webkit-scrollbar-track {
          background: transparent;
          border-radius: 10px;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb {
          background-color: #ccc;
          border-radius: 10px;
          border: 2px solid transparent;
          background-clip: content-box;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb:hover {
          background-color: #999;
        }

        .custom-scrollbar {
          scrollbar-width: thin;
          scrollbar-color: #ccc transparent;
        }
      ` }), _jsx("div", Object.assign({ style: {
                    position: "fixed",
                    bottom: "20px",
                    right: "20px",
                    width: "60px",
                    height: "60px",
                    backgroundColor: "#007bff",
                    borderRadius: "50%",
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                    cursor: "pointer",
                    zIndex: 9999,
                    color: "#fff",
                    fontSize: "28px",
                    boxShadow: "0 4px 5px rgba(0, 0, 0, 0.3)",
                }, onClick: () => setIsOpen(!isOpen), title: "Open Auth Widget" }, { children: "\uD83D\uDCAC" })), isOpen && (_jsx("div", Object.assign({ className: "custom-scrollbar", style: {
                    position: "fixed",
                    bottom: "90px",
                    right: "20px",
                    width: "68rem",
                    maxHeight: "75vh",
                    zIndex: 9998,
                    overflowY: "auto",
                } }, { children: error ? (_jsx("div", Object.assign({ style: {
                        paddingTop: "200px",
                        minHeight: "500px",
                    } }, { children: _jsx(ErrorWidget, { message: error, onClose: () => setIsOpen(false) }) }))) : (_jsxs("div", Object.assign({ style: {
                        maxWidth: 1200,
                        margin: "auto",
                        border: "1px solid #ccc",
                        borderRadius: 16,
                        overflow: "hidden",
                        backgroundColor: "#ffffff",
                    } }, { children: [_jsxs("div", Object.assign({ style: {
                                backgroundColor: "#F1F7FB",
                                padding: "16px 24px",
                                display: "flex",
                                justifyContent: "space-between",
                                alignItems: "center",
                                borderBottom: "1px solid #BFDBFE",
                            } }, { children: [_jsx("h2", Object.assign({ style: {
                                        fontSize: 20,
                                        fontWeight: 500,
                                        color: "#042567",
                                        margin: 0,
                                    } }, { children: "Potential Deductions Analysis" })), _jsx("div", Object.assign({ style: { display: "flex", alignItems: "center", gap: 10 } }, { children: _jsxs("span", Object.assign({ style: {
                                            fontSize: 14,
                                            color: "#6B7280",
                                            display: "flex",
                                            alignItems: "center",
                                            gap: 4,
                                        } }, { children: [_jsx("span", { style: {
                                                    width: 8,
                                                    height: 8,
                                                    borderRadius: "50%",
                                                    backgroundColor: "#687DF7",
                                                    display: "inline-block",
                                                } }), "Analysis complete"] })) }))] })), !selected && (_jsx("div", Object.assign({ style: {
                                display: "flex",
                                alignItems: "center",
                                justifyContent: "end",
                                padding: "0px 10px",
                                marginBottom: "18px",
                                marginTop: "18px",
                            } }, { children: _jsxs("select", Object.assign({ id: "yearSelect", value: year || "", onChange: (e) => setYear(parseInt(e.target.value)), style: {
                                    padding: "8px",
                                    borderRadius: "6px",
                                    border: "1px solid rgba(30, 64, 175, 0.1)",
                                    width: "100%",
                                    outline: "none",
                                    background: "#ffffff",
                                    maxWidth: "200px",
                                    marginTop: "5px",
                                } }, { children: [_jsx("option", Object.assign({ value: "", disabled: true }, { children: "Select Year" })), (() => {
                                        const currentYear = new Date().getFullYear();
                                        const years = Array.from({ length: currentYear }, (_, i) => currentYear - i);
                                        return years.map((y) => (_jsx("option", Object.assign({ value: y }, { children: y }), y)));
                                    })()] })) }))), " ", _jsx(_Fragment, { children: year && (_jsxs(_Fragment, { children: [_jsx("div", Object.assign({ style: { padding: "0 24px 10px" } }, { children: loading ? (_jsx("div", Object.assign({ style: {
                                                display: "flex",
                                                justifyContent: "center",
                                                alignItems: "center",
                                                fontSize: "14px",
                                                fontWeight: "normal",
                                                color: "#333333",
                                                marginTop: "40px",
                                                marginBottom: "40px",
                                            } }, { children: "loading.." }))) : (_jsx(_Fragment, { children: _jsxs("div", Object.assign({ style: { padding: "0px" } }, { children: [_jsx("div", Object.assign({ style: {
                                                            display: "flex",
                                                            alignItems: "center",
                                                            gap: "10px",
                                                            marginBottom: "10px",
                                                            flexWrap: "wrap",
                                                            justifyContent: "start",
                                                            marginTop: "10px",
                                                        } }, { children: (filteredSubcategories === null || filteredSubcategories === void 0 ? void 0 : filteredSubcategories.length) === 0 ? ("") : (_jsx(_Fragment, { children: !selected && (_jsx(_Fragment, { children: Object.entries((transactions === null || transactions === void 0 ? void 0 : transactions.categories) || {}).map(([catKey, catValue]) => (_jsx("button", Object.assign({ onClick: () => setSelectedCategoryKey(selectedCategoryKey === catKey
                                                                        ? null
                                                                        : catKey), style: {
                                                                        border: "1px solid #5387F1",
                                                                        padding: "12px 16px",
                                                                        fontSize: 14,
                                                                        fontWeight: 500,
                                                                        borderRadius: 8,
                                                                        backgroundColor: selectedCategoryKey === catKey
                                                                            ? "#5387F1"
                                                                            : "white",
                                                                        color: selectedCategoryKey === catKey
                                                                            ? "white"
                                                                            : "#5387F1",
                                                                        cursor: "pointer",
                                                                    } }, { children: catValue === null || catValue === void 0 ? void 0 : catValue.form }), catKey))) })) })) })), !selected ? (filtered.length === 0 ? (_jsx("p", Object.assign({ style: {
                                                            fontSize: "0.875rem",
                                                            color: "#526282",
                                                            margin: 0,
                                                            display: "flex",
                                                            justifyContent: "center",
                                                            width: "100%",
                                                        } }, { children: "No transactions match." }))) : (filtered.map(([key, g]) => {
                                                        var _a, _b, _c, _d, _e;
                                                        return (_jsxs("div", Object.assign({ onClick: () => setSelected(g), style: {
                                                                cursor: "pointer",
                                                                display: "flex",
                                                                justifyContent: "space-between",
                                                                alignItems: "center",
                                                                padding: "16px 20px",
                                                                border: "1px solid #1E40AF1A",
                                                                borderRadius: 12,
                                                                marginBottom: "12px",
                                                            } }, { children: [_jsxs("div", { children: [_jsx("h3", Object.assign({ style: {
                                                                                fontSize: "15px",
                                                                                fontWeight: 500,
                                                                                color: "#042567",
                                                                                margin: "0 0 6px 0",
                                                                            } }, { children: key !== null && key !== void 0 ? key : "NA" })), _jsx("span", Object.assign({ style: {
                                                                                fontSize: "0.75rem",
                                                                                backgroundColor: "#4A6CF71F",
                                                                                color: "#4A6CF7",
                                                                                borderRadius: 9999,
                                                                                padding: "4px 16px",
                                                                                display: "inline-block",
                                                                                border: "1px solid #4A6CF71A",
                                                                            } }, { children: ((_a = g === null || g === void 0 ? void 0 : g.parentCategoryName) !== null && _a !== void 0 ? _a : "") !== ""
                                                                                ? g === null || g === void 0 ? void 0 : g.parentCategoryName
                                                                                : "N/A" })), _jsxs("p", Object.assign({ style: {
                                                                                fontSize: "0.75rem",
                                                                                borderRadius: 9999,
                                                                            } }, { children: ["Based on ", (_c = (_b = g === null || g === void 0 ? void 0 : g.transactions) === null || _b === void 0 ? void 0 : _b.length) !== null && _c !== void 0 ? _c : 0, " ", "transaction"] }))] }), _jsxs("div", Object.assign({ style: { textAlign: "right" } }, { children: [_jsxs("p", Object.assign({ style: {
                                                                                color: "#042567",
                                                                                fontSize: "16px",
                                                                                fontWeight: 500,
                                                                                margin: "0 0 4px",
                                                                                display: "flex",
                                                                                alignItems: "center",
                                                                                gap: 6,
                                                                            } }, { children: ["$", formatNumberWithCommas((_d = g.total_amount.toFixed(2)) !== null && _d !== void 0 ? _d : 0), " "] })), _jsxs("p", Object.assign({ style: {
                                                                                fontSize: "0.875rem",
                                                                                color: "#526282",
                                                                                margin: 0,
                                                                            } }, { children: ["Potential:", " ", formatNumberWithCommas((_e = g.total_amount.toFixed(2)) !== null && _e !== void 0 ? _e : 0)] }))] }))] }), g.category));
                                                    }))) : (_jsx(_Fragment, { children: _jsxs("div", Object.assign({ style: {
                                                                backgroundColor: "#fff",
                                                                color: "#000",
                                                                borderRadius: "12px",
                                                                padding: "0px 0px 1rem",
                                                                maxWidth: "100%",
                                                                margin: "0px auto 1rem",
                                                                fontFamily: "sans-serif",
                                                                fontSize: "0.9rem",
                                                            } }, { children: [_jsxs("button", Object.assign({ onClick: () => {
                                                                        setSelected(null);
                                                                    }, style: {
                                                                        background: "transparent",
                                                                        border: "none",
                                                                        color: "#2094EA",
                                                                        marginBottom: "1rem",
                                                                        cursor: "pointer",
                                                                        fontWeight: 500,
                                                                        fontSize: "1rem",
                                                                        display: "flex",
                                                                        alignItems: "center",
                                                                        gap: "5px",
                                                                    } }, { children: [_jsx(HiOutlineArrowSmLeft, { style: { fontSize: "22px" } }), "Back"] })), _jsxs("div", Object.assign({ style: {
                                                                        padding: "0px",
                                                                    } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                                backgroundColor: "#F1F7FB",
                                                                                borderRadius: "8px",
                                                                                padding: "1rem",
                                                                                display: "flex",
                                                                                justifyContent: "space-between",
                                                                                alignItems: "center",
                                                                                marginBottom: "14px",
                                                                                flexWrap: "wrap",
                                                                            } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                                        display: "flex",
                                                                                        flexDirection: "column",
                                                                                    } }, { children: [_jsx("h2", Object.assign({ style: {
                                                                                                color: "#042567",
                                                                                                fontSize: "16px",
                                                                                                fontWeight: "500",
                                                                                                margin: 0,
                                                                                                textTransform: "capitalize",
                                                                                            } }, { children: selected === null || selected === void 0 ? void 0 : selected.name })), _jsxs("span", Object.assign({ style: {
                                                                                                backgroundColor: "#67DC791F",
                                                                                                color: "#15803D",
                                                                                                padding: "6px 16px",
                                                                                                border: "1px solid #4EC7601A",
                                                                                                borderRadius: "50px",
                                                                                                fontSize: "0.85rem",
                                                                                                marginTop: "6px",
                                                                                                textAlign: "center",
                                                                                            } }, { children: ["Deduction Opportunity", " "] }))] })), _jsxs("div", Object.assign({ style: { textAlign: "right" } }, { children: [_jsxs("p", Object.assign({ style: {
                                                                                                color: "#042567",
                                                                                                fontSize: "17px",
                                                                                                fontWeight: "500",
                                                                                                margin: 0,
                                                                                            } }, { children: ["$", formatNumberWithCommas((_a = selected.total_amount.toFixed(2)) !== null && _a !== void 0 ? _a : 0)] })), _jsxs("span", Object.assign({ style: {
                                                                                                fontSize: "14px",
                                                                                                color: "#526282",
                                                                                            } }, { children: [selected.transactions.length, " ", "transactions"] }))] }))] })), _jsx("div", Object.assign({ style: {
                                                                                border: "1px solid #E4EEF4",
                                                                                borderRadius: "12px",
                                                                                overflow: "hidden",
                                                                            } }, { children: _jsxs("table", Object.assign({ style: {
                                                                                    width: "100%",
                                                                                    borderCollapse: "collapse",
                                                                                    background: "#fff",
                                                                                    borderRadius: "8px",
                                                                                    overflow: "hidden",
                                                                                } }, { children: [_jsx("thead", Object.assign({ style: {
                                                                                            backgroundColor: "#f1f5f9",
                                                                                            textAlign: "left",
                                                                                        } }, { children: _jsxs("tr", { children: [_jsx("th", Object.assign({ style: {
                                                                                                        padding: "12px 16px",
                                                                                                        color: "#042567",
                                                                                                        textAlign: "left",
                                                                                                        fontWeight: "500",
                                                                                                    } }, { children: "Description" })), _jsx("th", Object.assign({ style: {
                                                                                                        padding: "12px 16px",
                                                                                                        color: "#042567",
                                                                                                        textAlign: "left",
                                                                                                        fontWeight: "500",
                                                                                                    } }, { children: "Type" })), _jsx("th", Object.assign({ style: {
                                                                                                        padding: "12px 16px",
                                                                                                        color: "#042567",
                                                                                                        textAlign: "left",
                                                                                                        fontWeight: "500",
                                                                                                    } }, { children: "Amount" })), _jsx("th", Object.assign({ style: {
                                                                                                        padding: "12px 16px",
                                                                                                        color: "#042567",
                                                                                                        textAlign: "left",
                                                                                                        fontWeight: "500",
                                                                                                    } }, { children: "Confidence Score" })), _jsx("th", Object.assign({ style: {
                                                                                                        padding: "12px 16px",
                                                                                                        color: "#042567",
                                                                                                        textAlign: "left",
                                                                                                        fontWeight: "500",
                                                                                                    } }, { children: "Category" })), _jsx("th", Object.assign({ style: {
                                                                                                        padding: "12px 16px",
                                                                                                        color: "#042567",
                                                                                                        textAlign: "left",
                                                                                                        fontWeight: "500",
                                                                                                    } }, { children: "Date" }))] }) })), _jsx("tbody", { children: (_b = selected === null || selected === void 0 ? void 0 : selected.transactions) === null || _b === void 0 ? void 0 : _b.map((t) => {
                                                                                            var _a, _b, _c, _d, _e, _f, _g;
                                                                                            const transactionId = (t === null || t === void 0 ? void 0 : t.transaction_id) ||
                                                                                                (t === null || t === void 0 ? void 0 : t.description);
                                                                                            return (_jsx(React.Fragment, { children: _jsxs("tr", Object.assign({ style: {
                                                                                                        borderBottom: "1px solid #e0e0e0",
                                                                                                    } }, { children: [_jsx("td", Object.assign({ style: {
                                                                                                                padding: "12px 16px",
                                                                                                                color: "#444",
                                                                                                                textAlign: "left",
                                                                                                            } }, { children: (_a = t.name) !== null && _a !== void 0 ? _a : t.merchant })), _jsx("td", Object.assign({ style: {
                                                                                                                padding: "12px 16px",
                                                                                                                textAlign: "left",
                                                                                                            } }, { children: ((_b = t.schedule) !== null && _b !== void 0 ? _b : "") !== ""
                                                                                                                ? t.schedule
                                                                                                                : ((_c = selected === null || selected === void 0 ? void 0 : selected.schedule) !== null && _c !== void 0 ? _c : "") !== ""
                                                                                                                    ? selected.schedule
                                                                                                                    : "N/A" })), _jsx("td", Object.assign({ style: {
                                                                                                                padding: "12px 16px",
                                                                                                                textAlign: "left",
                                                                                                                color: t.amount < 0
                                                                                                                    ? "#d32f2f"
                                                                                                                    : "#2e7d32",
                                                                                                            } }, { children: formatNumberWithCommas((_e = (_d = t === null || t === void 0 ? void 0 : t.amount) === null || _d === void 0 ? void 0 : _d.toFixed(2)) !== null && _e !== void 0 ? _e : 0) })), _jsx("td", Object.assign({ style: {
                                                                                                                padding: "12px 16px",
                                                                                                                textAlign: "left",
                                                                                                            } }, { children: (_f = t === null || t === void 0 ? void 0 : t.confidence_score) !== null && _f !== void 0 ? _f : 0 })), _jsx("td", Object.assign({ style: {
                                                                                                                padding: "12px 16px",
                                                                                                                textAlign: "left",
                                                                                                            } }, { children: (_g = t === null || t === void 0 ? void 0 : t.category) !== null && _g !== void 0 ? _g : "N/A" })), _jsx("td", Object.assign({ style: {
                                                                                                                padding: "12px 16px",
                                                                                                                textAlign: "left",
                                                                                                            } }, { children: (t === null || t === void 0 ? void 0 : t.date)
                                                                                                                ? new Date(t.date).toLocaleDateString("en-US", {
                                                                                                                    year: "numeric",
                                                                                                                    month: "short",
                                                                                                                    day: "numeric",
                                                                                                                })
                                                                                                                : "N/A" }))] })) }, transactionId));
                                                                                        }) })] })) }))] }))] })) }))] })) })) })), !selected &&
                                        filteredSubcategories.length > 0 &&
                                        !loading && (_jsx("div", Object.assign({ style: { padding: "0 24px 24px" } }, { children: _jsxs("div", Object.assign({ style: {
                                                backgroundColor: "#5387F1",
                                                padding: "20px 24px",
                                                borderRadius: 16,
                                                color: "white",
                                                textAlign: "center",
                                                marginTop: 0,
                                                maxWidth: "400px",
                                                marginLeft: "auto",
                                                marginRight: "auto",
                                            } }, { children: [_jsxs("p", Object.assign({ style: {
                                                        fontSize: 16,
                                                        fontWeight: 500,
                                                        color: "#ffffff",
                                                        margin: 4,
                                                    } }, { children: ["Total potential deduction: $", formatNumberWithCommas((_e = (_d = (_c = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _c === void 0 ? void 0 : _c.total_deductible_amount) === null || _d === void 0 ? void 0 : _d.toFixed(2)) !== null && _e !== void 0 ? _e : 0)] })), _jsxs("p", Object.assign({ style: {
                                                        fontSize: 14,
                                                        margin: 0,
                                                        color: "#ffffff",
                                                    } }, { children: ["Total amount: $", formatNumberWithCommas((_h = (_g = (_f = transactions === null || transactions === void 0 ? void 0 : transactions.summary) === null || _f === void 0 ? void 0 : _f.potential_savings) === null || _g === void 0 ? void 0 : _g.toFixed(2)) !== null && _h !== void 0 ? _h : 0)] })), _jsx("button", Object.assign({ style: {
                                                        marginTop: 16,
                                                        padding: "8px 16px",
                                                        borderRadius: 8,
                                                        backgroundColor: "white",
                                                        color: "#5387F1",
                                                        border: "none",
                                                        fontWeight: 500,
                                                        cursor: "pointer",
                                                    } }, { children: "Get Insights" }))] })) })))] })) })] }))) })))] }));
};
export default PotentialDeductions;
