// @rohit23061999/transaction-review-widget/dist/historicalAnalysisWidget/historicalAnalysis.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import { useState, useRef, useEffect } from "react";
import Chart from "chart.js/auto";
import { toast } from "react-toastify";
import { getAuthSessionToken } from "../helper/AuthHelper";
import ErrorWidget from "../helper/Error";
import api from "../api/api";
import { formatNumberWithCommas } from "../helper/NumberFormat";
const { AiOutlineCloudUpload } = require("react-icons/ai");
const HistoricalAnalysisWidget = ({ userId, access_token, session_token, onError, }) => {
    var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p;
    const [loading, setLoading] = useState(false);
    const [widgetloading, setWidgetLoading] = useState(false);
    const [year, setYear] = useState("");
    const [error, setError] = useState("");
    const [isOpen, setIsOpen] = useState(false);
    const [data, setData] = useState([]);
    const [showAnalysis, setShowAnalysis] = useState(false);
    const [activeTab, setActiveTab] = useState("previous");
    const [view, setView] = useState("Yearly");
    const chartRef = useRef(null);
    const transId = useRef("");
    const [maritalStatus, setMaritalStatus] = useState("Single"); // default value
    // 1️⃣  ───────────────────────────────────────────────────────────────────────
    // Put this near the top of your component:
    const [file, setFile] = useState(null);
    const [uploading, setUploading] = useState(false);
    // Inside your component
    const handleUpload = () => __awaiter(void 0, void 0, void 0, function* () {
        var _q, _r, _s;
        console.log(!year, "check eayr cond", year);
        if (!year) {
            toast.error("Please add year", { toastId: "year added" });
            return;
        }
        if (!file) {
            toast.error("Please select a file.", { toastId: "select-file" });
            return;
        }
        if (!maritalStatus) {
            toast.error("Please select a filling status", {
                toastId: "select-filess",
            });
            return;
        }
        setUploading(true);
        try {
            const formData = new FormData();
            formData.append("user_id", userId); // string
            formData.append("tax_year", year);
            formData.append("tax_return_file", file); // the PDF File object
            formData.append("filing_status", maritalStatus);
            // Axios will add the correct multipart boundary.
            const response = yield api.post("/v2/api/tax-return/upload", formData, {
                headers: { "Content-Type": "multipart/form-data" },
            });
            console.log(response, "responseresponseresponse");
            if ((response === null || response === void 0 ? void 0 : response.status) == 200) {
                toast.success("File uploaded successfully.");
                transId.current = (_q = response === null || response === void 0 ? void 0 : response.data) === null || _q === void 0 ? void 0 : _q.tax_return_id;
                yield handleApiCalling(false);
            }
            else {
                setShowAnalysis(false);
            }
            setShowAnalysis(true);
        }
        catch (err) {
            toast.error((_s = (_r = err === null || err === void 0 ? void 0 : err.response) === null || _r === void 0 ? void 0 : _r.data) === null || _s === void 0 ? void 0 : _s.detail, { toastId: "erur" });
        }
        finally {
            setUploading(false);
        }
    });
    const handleApiCalling = (type) => __awaiter(void 0, void 0, void 0, function* () {
        setShowAnalysis(true);
        setWidgetLoading(true);
        try {
            // Build params conditionally
            const params = { user_id: userId };
            if (!type)
                params.tax_year = year;
            const response = yield api.get(`/v2/api/tax-return/${transId.current}/findings`, { params });
            console.log(response, "responseresponseresponse");
            if ((response === null || response === void 0 ? void 0 : response.status) == 200) {
                setData(response === null || response === void 0 ? void 0 : response.data);
            }
            else {
                setWidgetLoading(false);
            }
        }
        catch (error) {
            setWidgetLoading(false);
            console.error("API error:", error);
            toast.error("Failed to fetch tax return data");
        }
        finally {
            setWidgetLoading(false);
        }
    });
    const handleApiCallingSkip = (type) => __awaiter(void 0, void 0, void 0, function* () {
        var _t;
        setShowAnalysis(true);
        setWidgetLoading(true);
        try {
            // Build params conditionally
            const params = { user_id: userId };
            if (!type)
                params.tax_year = year;
            const response = yield api.get(`/V2/api/tax-analysis/forms-needed`, {
                params,
            });
            console.log(response, "responseresponseresponse");
            if ((response === null || response === void 0 ? void 0 : response.status) == 200) {
                transId.current = (_t = response === null || response === void 0 ? void 0 : response.data) === null || _t === void 0 ? void 0 : _t.tax_return_id;
                yield handleApiCalling(true);
            }
            else {
                setWidgetLoading(false);
            }
        }
        catch (error) {
            setWidgetLoading(false);
            console.error("API error:", error);
            toast.error("Failed to fetch tax return data");
        }
        finally {
            setWidgetLoading(false);
        }
    });
    useEffect(() => {
        if (access_token && session_token) {
            getAuthSessionToken({
                session_token,
                userId,
                onSuccess: (data) => {
                    console.log("Auth token fetched, domain URLs:", data.domain_urls);
                },
                onError,
                setLoading,
                setError,
            });
        }
    }, [access_token, session_token]);
    useEffect(() => {
        if (showAnalysis && activeTab === "comparison" && chartRef.current) {
            new Chart(chartRef.current, {
                type: "bar",
                data: {
                    labels: [
                        "2019",
                        "2020",
                        "2021",
                        "2022",
                        "2023",
                        "2024",
                        "2025",
                        "Potential",
                    ],
                    datasets: [
                        {
                            label: "Total Deductions ($)",
                            data: [1800, 2000, 1500, 2500, 4000, 3500, 2900, 4200],
                            backgroundColor: "#72C9E7",
                            borderRadius: 6,
                            barThickness: 20,
                        },
                    ],
                },
                options: {
                    responsive: true,
                    plugins: {
                        legend: { display: false },
                        tooltip: { mode: "index", intersect: false },
                    },
                    scales: {
                        y: { beginAtZero: true },
                    },
                },
            });
        }
    }, [showAnalysis, activeTab]);
    const [viewMode, setViewMode] = useState("list");
    const [selectedDeduction, setSelectedDeduction] = useState(null);
    console.log(data, "datadatadatadata");
    return (_jsxs(_Fragment, { children: [_jsx("style", { children: `
        .custom-scrollbar::-webkit-scrollbar {
          width: 8px;
        }

        .custom-scrollbar::-webkit-scrollbar-track {
          background: transparent;
          border-radius: 10px;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb {
          background-color: #ccc;
          border-radius: 10px;
          border: 2px solid transparent;
          background-clip: content-box;
        }

        .custom-scrollbar::-webkit-scrollbar-thumb:hover {
          background-color: #999;
        }

        .custom-scrollbar {
          scrollbar-width: thin;
          scrollbar-color: #ccc transparent;
        }
      ` }), _jsx("div", Object.assign({ style: {
                    position: "fixed",
                    bottom: "20px",
                    right: "20px",
                    width: "60px",
                    height: "60px",
                    backgroundColor: "#007bff",
                    borderRadius: "50%",
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                    cursor: "pointer",
                    zIndex: 9999,
                    color: "#fff",
                    fontSize: "28px",
                    boxShadow: "0 4px 5px rgba(0, 0, 0, 0.3)",
                }, onClick: () => setIsOpen(!isOpen), title: "Open Auth Widget" }, { children: "\uD83D\uDCAC" })), isOpen && (_jsx(_Fragment, { children: loading ? (_jsx("p", Object.assign({ style: {
                        fontSize: "0.875rem",
                        color: "#526282",
                        margin: "40px 0px",
                        display: "flex",
                        justifyContent: "center",
                        width: "100%",
                    } }, { children: "Loading\u2026" }))) : error ? (_jsx("div", Object.assign({ style: {
                        padding: "0",
                        minHeight: "500px",
                        paddingTop: "200px",
                    } }, { children: _jsx(ErrorWidget, { message: error, onClose: () => setIsOpen(false) }) }))) : (_jsx("div", Object.assign({ className: "custom-scrollbar", style: {
                        position: "fixed",
                        bottom: "90px",
                        right: "20px",
                        width: "68rem",
                        maxHeight: "75vh",
                        zIndex: 9998,
                        padding: "0",
                        overflowY: "auto",
                    } }, { children: _jsx("div", Object.assign({ style: {
                            maxWidth: "1200px",
                            border: "1px solid #ccc",
                            borderRadius: "16px",
                            overflow: "hidden",
                            backgroundColor: "#ffffff",
                        } }, { children: !showAnalysis ? (_jsxs("div", { children: [_jsx("div", Object.assign({ style: { background: "#F1F7FB", padding: "20px 30px" } }, { children: _jsx("h2", Object.assign({ style: {
                                            margin: 0,
                                            fontSize: "22px",
                                            fontWeight: 600,
                                            color: "#042567",
                                        } }, { children: "Upload Your Tax Return" })) })), _jsxs("div", Object.assign({ style: {
                                        marginBottom: 0,
                                        display: "flex",
                                        justifyContent: "space-between",
                                        alignItems: "center",
                                        padding: "18px 30px 5px",
                                        gap: "20px",
                                    } }, { children: [_jsx("label", Object.assign({ htmlFor: "yearInput", style: {
                                                display: "block",
                                                marginBottom: 0,
                                                color: "#042567",
                                                padding: "0px",
                                            } }, { children: "Enter Tax Year (Current or Previous Only)" })), _jsxs("div", Object.assign({ style: {
                                                display: "flex",
                                                alignItems: "center",
                                                justifyContent: "end",
                                                gap: "10px",
                                            } }, { children: [_jsxs("select", Object.assign({ id: "yearSelect", value: year || "", onChange: (e) => setYear(parseInt(e.target.value)), style: {
                                                        padding: "8px",
                                                        borderRadius: "6px",
                                                        border: "1px solid rgba(30, 64, 175, 0.1)",
                                                        width: "140px",
                                                        outline: "none",
                                                        background: "#ffffff",
                                                    } }, { children: [_jsx("option", Object.assign({ value: "", disabled: true }, { children: "Select Year" })), (() => {
                                                            const currentYear = new Date().getFullYear();
                                                            const years = Array.from({ length: currentYear }, (_, i) => currentYear - i);
                                                            return years.map((y) => (_jsx("option", Object.assign({ value: y }, { children: y }), y)));
                                                        })()] })), _jsx("div", { children: _jsxs("select", Object.assign({ value: maritalStatus, onChange: (e) => setMaritalStatus(e.target.value), style: {
                                                            padding: "8px",
                                                            borderRadius: "6px",
                                                            border: "1px solid rgba(30, 64, 175, 0.1)",
                                                            outline: "none",
                                                            background: "#ffffff",
                                                            width: "120px",
                                                        } }, { children: [_jsx("option", Object.assign({ value: "Single" }, { children: "Single" })), _jsx("option", Object.assign({ value: "Married" }, { children: "Married" }))] })) })] }))] })), _jsxs("div", Object.assign({ style: { padding: "20px 30px 30px" } }, { children: [_jsxs("div", Object.assign({ style: {
                                                border: "1px solid #ccc",
                                                padding: 20,
                                                borderRadius: 12,
                                                marginBottom: 20,
                                            } }, { children: [_jsx("h3", Object.assign({ style: {
                                                        color: "#042567",
                                                        fontSize: 20,
                                                        fontWeight: 500,
                                                        marginBottom: "8px",
                                                    } }, { children: "Upload file" })), _jsxs("div", Object.assign({ style: {
                                                        border: "1px dashed #ccc",
                                                        padding: 30,
                                                        textAlign: "center",
                                                        borderRadius: 10,
                                                    } }, { children: [_jsx("input", { type: "file", id: "fileInput", accept: ".pdf,.csv", style: { display: "none" }, onChange: (e) => {
                                                                var _a;
                                                                const picked = (_a = e.target.files) === null || _a === void 0 ? void 0 : _a[0];
                                                                if (!picked)
                                                                    return;
                                                                const isPDF = picked.type === "application/pdf" ||
                                                                    picked.name.endsWith(".pdf");
                                                                const isCSV = picked.type === "text/csv" ||
                                                                    picked.name.endsWith(".csv");
                                                                if (!isPDF && !isCSV) {
                                                                    toast.error("Only PDF or CSV files are allowed.");
                                                                    return;
                                                                }
                                                                setFile(picked);
                                                            } }), !file ? (_jsx("label", Object.assign({ htmlFor: "fileInput", style: { cursor: "pointer" } }, { children: _jsxs("div", Object.assign({ style: {
                                                                    display: "flex",
                                                                    flexDirection: "column",
                                                                    alignItems: "center",
                                                                    gap: "4px",
                                                                } }, { children: [_jsx(AiOutlineCloudUpload, { style: { color: "#1495EA", fontSize: 22 } }), _jsx("strong", Object.assign({ style: { color: "#042567", fontSize: 16 } }, { children: "Upload CSV File" })), _jsx("span", Object.assign({ style: { color: "#526282", fontSize: 14 } }, { children: "Drag-and-drop upload zone for PDF/CSV/Tax software export" }))] })) }))) : (_jsxs("div", Object.assign({ style: {
                                                                display: "flex",
                                                                alignItems: "start",
                                                                justifyContent: "center",
                                                                position: "relative",
                                                            } }, { children: [_jsxs("p", { children: [file.name, " (", (_a = (file.size / 1024)) === null || _a === void 0 ? void 0 : _a.toFixed(2), " ", "KB)"] }), _jsx("button", Object.assign({ disabled: uploading, type: "button", onClick: () => setFile(null), style: {
                                                                        backgroundColor: "#ffffff",
                                                                        border: "1px solid rgb(229 29 65)",
                                                                        color: "rgb(229 29 65)",
                                                                        width: "20px",
                                                                        height: "20px",
                                                                        padding: "2px",
                                                                        borderRadius: "50px",
                                                                        cursor: "pointer",
                                                                        display: "flex",
                                                                        alignItems: "center",
                                                                        justifyContent: "center",
                                                                    } }, { children: _jsxs("svg", Object.assign({ stroke: "currentColor", fill: "currentColor", "stroke-width": "0", viewBox: "0 0 24 24", height: "1em", width: "1em", xmlns: "http://www.w3.org/2000/svg" }, { children: [_jsx("path", { fill: "none", d: "M0 0h24v24H0z" }), _jsx("path", { d: "M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" })] })) }))] })))] }))] })), _jsxs("div", Object.assign({ style: {
                                                border: "1px solid #ccc",
                                                padding: 20,
                                                borderRadius: 12,
                                            } }, { children: [_jsx("h3", Object.assign({ style: {
                                                        color: "#042567",
                                                        fontSize: 20,
                                                        fontWeight: 500,
                                                    } }, { children: "Why upload your return?" })), _jsx("div", Object.assign({ style: { display: "flex", gap: 20, marginTop: 25 } }, { children: [
                                                        {
                                                            text: "Instantly analyze past overpayments",
                                                            desc: "We’ll scan for missed deductions and overpaid taxes.",
                                                            icon: "https://i.ibb.co/Kj94Fgz8/deduction-icon.png",
                                                        },
                                                        {
                                                            text: "Build a smarter plan for this year",
                                                            desc: "Get recommendations to optimize this year’s return.",
                                                            icon: "https://i.ibb.co/MDVspRq8/smarter-icon.png",
                                                        },
                                                        {
                                                            text: "Personalized refund suggestions",
                                                            desc: "We’ll show where you can save based on your data.",
                                                            icon: "https://i.ibb.co/C3MQr5NC/suggestion-icon.png",
                                                        },
                                                    ].map(({ text, desc, icon }, idx) => (_jsxs("div", Object.assign({ style: { flex: 1 } }, { children: [_jsxs("div", Object.assign({ style: { marginBottom: 12 } }, { children: [" ", _jsx("img", { src: icon, width: 50, height: 50, alt: "" })] })), _jsx("p", Object.assign({ style: {
                                                                    color: "#042567",
                                                                    fontWeight: 500,
                                                                    margin: "0px 0px 8px 0px",
                                                                } }, { children: text })), _jsx("p", Object.assign({ style: { color: "#526282", fontSize: 14 } }, { children: desc }))] }), idx))) }))] })), _jsxs("div", Object.assign({ style: { textAlign: "right", marginTop: 20 } }, { children: [_jsx("button", Object.assign({ onClick: handleUpload, disabled: uploading, style: {
                                                        marginRight: 10,
                                                        padding: "10px 20px",
                                                        border: "1px solid #5387F1",
                                                        background: "#ffffff",
                                                        color: "#5387F1",
                                                        borderRadius: 8,
                                                        cursor: "pointer",
                                                    } }, { children: uploading ? " Uploading..." : "Upload File" })), _jsx("button", Object.assign({ type: "button", disabled: uploading, onClick: () => handleApiCallingSkip(true), style: {
                                                        padding: "10px 20px",
                                                        background: "#5387F1",
                                                        color: "white",
                                                        border: "none",
                                                        borderRadius: 8,
                                                        cursor: "pointer",
                                                    } }, { children: "Skip This Step" }))] }))] }))] })) : (_jsxs("div", { children: [_jsx("div", Object.assign({ style: { background: "#F1F7FB", padding: "20px 30px" } }, { children: _jsx("h2", Object.assign({ style: {
                                            margin: 0,
                                            fontSize: 20,
                                            fontWeight: 600,
                                            color: "#042567",
                                        } }, { children: "Historical Analysis Widget" })) })), widgetloading ? (_jsx("p", Object.assign({ style: {
                                        fontSize: "0.875rem",
                                        color: "#526282",
                                        margin: "20px 0px",
                                        display: "flex",
                                        justifyContent: "center",
                                        width: "100%",
                                    } }, { children: "Loading\u2026" }))) : (_jsxs("div", Object.assign({ style: { padding: "14px 30px" } }, { children: [_jsx("p", Object.assign({ style: {
                                                color: "#526282",
                                                fontSize: 16,
                                                marginTop: "0px",
                                                marginBottom: "16px",
                                            } }, { children: "Analyses previous tax returns to identify missed deduction opportunities" })), _jsx("div", Object.assign({ style: { margin: "20px 0" } }, { children: _jsx("button", Object.assign({ onClick: () => setActiveTab("previous"), style: {
                                                    padding: "10px 20px",
                                                    marginRight: 10,
                                                    borderRadius: 8,
                                                    background: activeTab === "previous" ? "#5387F1" : "white",
                                                    color: activeTab === "previous" ? "white" : "#5387F1",
                                                    border: "1px solid #5387F1",
                                                    cursor: "pointer",
                                                } }, { children: "Previous Return Analysis" })) })), activeTab === "previous" ? (_jsx("div", { children: _jsxs("div", Object.assign({ style: {
                                                    backgroundColor: "#F1F7FB",
                                                    border: "1px solid rgba(30, 64, 175, 0.1)",
                                                    borderRadius: 12,
                                                    padding: 20,
                                                } }, { children: [_jsxs("div", Object.assign({ style: {
                                                            display: "flex",
                                                            alignItems: "center",
                                                            gap: 4,
                                                            marginBottom: 12,
                                                        } }, { children: [_jsx("span", Object.assign({ style: {
                                                                    color: "#042567",
                                                                    fontSize: 18,
                                                                    fontWeight: 500,
                                                                } }, { children: "Previous Return Analysis" })), _jsxs("span", Object.assign({ style: {
                                                                    color: "#526282",
                                                                    fontSize: 16,
                                                                    fontWeight: 400,
                                                                } }, { children: ["(", year, " Tax Return Analyzed)"] }))] })), _jsx("div", Object.assign({ style: {
                                                            display: "grid",
                                                            gridTemplateColumns: "1fr 1fr",
                                                            gap: 16,
                                                        } }, { children: viewMode === "list" ? (_jsxs(_Fragment, { children: [_jsxs("div", Object.assign({ style: {
                                                                        background: "#ffffff",
                                                                        padding: 20,
                                                                        borderRadius: 12,
                                                                        border: "1px solid rgba(30,64,175,0.1)",
                                                                    } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                                display: "flex",
                                                                                alignItems: "center",
                                                                                gap: 6,
                                                                                marginBottom: 16,
                                                                            } }, { children: [_jsxs("svg", Object.assign({ xmlns: "http://www.w3.org/2000/svg", width: "22", height: "23", viewBox: "0 0 24 25", fill: "none" }, { children: [_jsx("path", { d: "M22 11.58V12.5C21.9988 14.6564 21.3005 16.7547 20.0093 18.4818C18.7182 20.209 16.9033 21.4725 14.8354 22.0839C12.7674 22.6953 10.5573 22.6219 8.53447 21.8746C6.51168 21.1273 4.78465 19.7461 3.61096 17.9371C2.43727 16.128 1.87979 13.9881 2.02168 11.8363C2.16356 9.68455 2.99721 7.63631 4.39828 5.99706C5.79935 4.35781 7.69279 3.21537 9.79619 2.74013C11.8996 2.2649 14.1003 2.48232 16.07 3.35999", stroke: "#4EC760", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }), _jsx("path", { d: "M22 4.5L12 14.51L9 11.51", stroke: "#4EC760", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" })] })), _jsx("h2", Object.assign({ style: {
                                                                                        fontSize: 18,
                                                                                        fontWeight: 500,
                                                                                        color: "#3F4540",
                                                                                        margin: 0,
                                                                                    } }, { children: "Claimed Deductions" }))] })), data && ((_b = data === null || data === void 0 ? void 0 : data.findings) === null || _b === void 0 ? void 0 : _b.length) > 0 ? ((_c = data === null || data === void 0 ? void 0 : data.findings) === null || _c === void 0 ? void 0 : _c.map((label, idx) => {
                                                                            var _a, _b;
                                                                            return (_jsxs("div", Object.assign({ onClick: () => {
                                                                                    setSelectedDeduction(label === null || label === void 0 ? void 0 : label.supporting_transactions);
                                                                                    setViewMode("details");
                                                                                }, style: {
                                                                                    cursor: "pointer",
                                                                                    background: "#EDFBEF",
                                                                                    borderLeft: "2px solid #4EC760",
                                                                                    borderRadius: 8,
                                                                                    display: "flex",
                                                                                    justifyContent: "space-between",
                                                                                    alignItems: "center",
                                                                                    padding: "12px 16px",
                                                                                    marginBottom: 12,
                                                                                } }, { children: [_jsxs("h4", Object.assign({ style: {
                                                                                            color: "#5C7861",
                                                                                            fontSize: 16,
                                                                                            fontWeight: 400,
                                                                                            margin: 0,
                                                                                        } }, { children: [label === null || label === void 0 ? void 0 : label.description, _jsxs("p", Object.assign({ style: {
                                                                                                    fontSize: 12,
                                                                                                    color: "#705E5E",
                                                                                                    fontStyle: "italic",
                                                                                                    margin: 0,
                                                                                                } }, { children: ["Line: ", label === null || label === void 0 ? void 0 : label.form_line] }))] })), _jsxs("span", Object.assign({ style: {
                                                                                            color: "#3F4540",
                                                                                            fontWeight: 500,
                                                                                            fontSize: 16,
                                                                                        } }, { children: ["$", formatNumberWithCommas((_b = (_a = label === null || label === void 0 ? void 0 : label.current_amount) === null || _a === void 0 ? void 0 : _a.toFixed(2)) !== null && _b !== void 0 ? _b : 0), " "] }))] }), idx));
                                                                        })) : (_jsx("span", { children: "No data found." })), _jsxs("div", Object.assign({ style: {
                                                                                background: "#F5FBFF",
                                                                                border: "1px solid #E0E7F6",
                                                                                borderRadius: 8,
                                                                                display: "flex",
                                                                                justifyContent: "space-between",
                                                                                alignItems: "center",
                                                                                padding: "12px 16px",
                                                                            } }, { children: [_jsx("h4", Object.assign({ style: {
                                                                                        color: "#5C7861",
                                                                                        fontSize: 16,
                                                                                        fontWeight: 500,
                                                                                        margin: 0,
                                                                                    } }, { children: "Total Claimed:" })), _jsxs("span", Object.assign({ style: {
                                                                                        color: "#3F4540",
                                                                                        fontWeight: 500,
                                                                                        fontSize: 16,
                                                                                    } }, { children: ["$", " ", (_f = formatNumberWithCommas((_e = (_d = data === null || data === void 0 ? void 0 : data.summary) === null || _d === void 0 ? void 0 : _d.potential_tax_savings) === null || _e === void 0 ? void 0 : _e.toFixed(2))) !== null && _f !== void 0 ? _f : 0] }))] }))] })), _jsxs("div", Object.assign({ style: {
                                                                        background: "#ffffff",
                                                                        padding: 20,
                                                                        borderRadius: 12,
                                                                        border: "1px solid rgba(115, 120, 134, 0.1)",
                                                                    } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                                display: "flex",
                                                                                alignItems: "center",
                                                                                gap: 6,
                                                                                marginBottom: 16,
                                                                            } }, { children: [_jsx("svg", Object.assign({ xmlns: "http://www.w3.org/2000/svg", width: "24", height: "25", viewBox: "0 0 24 25", fill: "none" }, { children: _jsx("path", { d: "M12 9.5V13.5M12 17.5H12.01M21.7299 18.5L13.7299 4.49998C13.5555 4.19218 13.3025 3.93617 12.9969 3.75805C12.6912 3.57993 12.3437 3.48608 11.9899 3.48608C11.6361 3.48608 11.2887 3.57993 10.983 3.75805C10.6773 3.93617 10.4244 4.19218 10.2499 4.49998L2.24993 18.5C2.07361 18.8053 1.98116 19.1519 1.98194 19.5045C1.98272 19.8571 2.07671 20.2032 2.25438 20.5078C2.43204 20.8124 2.68708 21.0646 2.99362 21.2388C3.30017 21.4131 3.64734 21.5032 3.99993 21.5H19.9999C20.3508 21.4996 20.6955 21.4069 20.9992 21.2313C21.303 21.0556 21.5551 20.8031 21.7304 20.4991C21.9057 20.1951 21.998 19.8504 21.9979 19.4995C21.9978 19.1486 21.9054 18.8039 21.7299 18.5Z", stroke: "#CC5555", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }) })), _jsx("h2", Object.assign({ style: {
                                                                                        fontSize: 18,
                                                                                        fontWeight: 500,
                                                                                        color: "#3F4540",
                                                                                        margin: 0,
                                                                                    } }, { children: "Missed Deductions" }))] })), data && ((_g = data === null || data === void 0 ? void 0 : data.findings) === null || _g === void 0 ? void 0 : _g.length) > 0 ? ((_h = data === null || data === void 0 ? void 0 : data.findings) === null || _h === void 0 ? void 0 : _h.map((item, idx) => {
                                                                            var _a, _b;
                                                                            return (_jsxs("div", Object.assign({ style: {
                                                                                    cursor: "pointer",
                                                                                    background: "#FFEBF0",
                                                                                    borderLeft: "2px solid #CC5555",
                                                                                    borderRadius: 8,
                                                                                    display: "flex",
                                                                                    justifyContent: "space-between",
                                                                                    alignItems: "center",
                                                                                    padding: "12px 16px",
                                                                                    marginBottom: 12,
                                                                                }, onClick: () => {
                                                                                    setSelectedDeduction(item === null || item === void 0 ? void 0 : item.supporting_transactions);
                                                                                    setViewMode("details");
                                                                                } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                                            display: "flex",
                                                                                            flexDirection: "column",
                                                                                        } }, { children: [_jsx("h4", Object.assign({ style: {
                                                                                                    color: "#503F3F",
                                                                                                    fontSize: 16,
                                                                                                    fontWeight: 400,
                                                                                                    margin: 0,
                                                                                                } }, { children: item === null || item === void 0 ? void 0 : item.description })), _jsxs("p", Object.assign({ style: {
                                                                                                    fontSize: 12,
                                                                                                    color: "#705E5E",
                                                                                                    fontStyle: "italic",
                                                                                                    margin: 0,
                                                                                                } }, { children: ["Line: ", item === null || item === void 0 ? void 0 : item.form_line] }))] })), _jsxs("span", Object.assign({ style: {
                                                                                            color: "#3F4540",
                                                                                            fontWeight: 500,
                                                                                            fontSize: 16,
                                                                                        } }, { children: ["$", formatNumberWithCommas((_b = (_a = item === null || item === void 0 ? void 0 : item.missed_amount) === null || _a === void 0 ? void 0 : _a.toFixed(2)) !== null && _b !== void 0 ? _b : 0)] }))] }), idx));
                                                                        })) : (_jsx("span", { children: "No data found" })), _jsxs("div", Object.assign({ style: {
                                                                                background: "#F5FBFF",
                                                                                border: "1px solid #E0E7F6",
                                                                                borderRadius: 8,
                                                                                display: "flex",
                                                                                justifyContent: "space-between",
                                                                                alignItems: "center",
                                                                                padding: "12px 16px",
                                                                            } }, { children: [_jsx("h4", Object.assign({ style: {
                                                                                        color: "#5C7861",
                                                                                        fontSize: 16,
                                                                                        fontWeight: 500,
                                                                                        margin: 0,
                                                                                    } }, { children: "Total Missed:" })), _jsxs("span", Object.assign({ style: {
                                                                                        color: "#3F4540",
                                                                                        fontWeight: 500,
                                                                                        fontSize: 16,
                                                                                    } }, { children: ["$", (_l = formatNumberWithCommas((_k = (_j = data === null || data === void 0 ? void 0 : data.summary) === null || _j === void 0 ? void 0 : _j.total_missed_amount) === null || _k === void 0 ? void 0 : _k.toFixed(2))) !== null && _l !== void 0 ? _l : 0] }))] }))] }))] })) : (_jsxs("div", { children: [_jsx("button", Object.assign({ onClick: () => setViewMode("list"), style: {
                                                                        marginBottom: 12,
                                                                        background: "#5387F1",
                                                                        color: "white",
                                                                        padding: "8px 16px",
                                                                        borderRadius: 8,
                                                                        border: "none",
                                                                    } }, { children: "\u2190 Back to Deductions" })), _jsx("h3", { children: selectedDeduction === null || selectedDeduction === void 0 ? void 0 : selectedDeduction.fieldName }), _jsxs("p", { children: ["Line: ", selectedDeduction === null || selectedDeduction === void 0 ? void 0 : selectedDeduction.scheduleCLine] }), _jsxs("p", { children: ["Amount: $", formatNumberWithCommas((_o = (_m = selectedDeduction === null || selectedDeduction === void 0 ? void 0 : selectedDeduction.amount) === null || _m === void 0 ? void 0 : _m.toFixed(2)) !== null && _o !== void 0 ? _o : 0)] }), _jsx("h4", { children: "Transactions:" }), ((_p = selectedDeduction === null || selectedDeduction === void 0 ? void 0 : selectedDeduction.transactions) === null || _p === void 0 ? void 0 : _p.length) ? (selectedDeduction.transactions.map((tx, idx) => {
                                                                    var _a, _b;
                                                                    return (_jsxs("div", Object.assign({ style: {
                                                                            border: "1px solid #ccc",
                                                                            borderRadius: 8,
                                                                            padding: "10px",
                                                                            marginBottom: 10,
                                                                        } }, { children: [_jsx("p", { children: _jsxs("strong", { children: ["Merchant:", tx.merchant] }) }), _jsx("p", { children: _jsxs("strong", { children: ["Category:", tx.category] }) }), _jsxs("p", { children: ["Name:", tx.name] }), _jsxs("p", { children: [" ", "Connfidence score:", tx.confidence_score] }), _jsxs("p", { children: ["Date:", new Date(tx.date).toLocaleDateString()] }), _jsxs("p", { children: ["Amount:$", formatNumberWithCommas((_b = (_a = tx.amount) === null || _a === void 0 ? void 0 : _a.toFixed(2)) !== null && _b !== void 0 ? _b : 0)] })] }), idx));
                                                                })) : (_jsx("p", { children: "No transactions found." }))] })) }))] })) })) : (_jsx("div", { children: _jsxs("div", Object.assign({ style: {
                                                    backgroundColor: "#F1F7FB",
                                                    border: "1px solid #1E40AF1A",
                                                    borderRadius: "0.75rem",
                                                    padding: "1.25rem",
                                                    marginBottom: "1.25rem",
                                                } }, { children: [_jsxs("div", Object.assign({ style: { marginBottom: "0.75rem" } }, { children: [_jsx("h4", Object.assign({ style: {
                                                                    color: "#042567",
                                                                    fontSize: "1.125rem",
                                                                    fontWeight: 500,
                                                                    marginBottom: "0.25rem",
                                                                    marginTop: 0,
                                                                } }, { children: "Year-over-Year Deduction Comparison" })), _jsx("p", Object.assign({ style: {
                                                                    color: "#526282",
                                                                    fontSize: "1rem",
                                                                    fontWeight: 400,
                                                                    margin: 0,
                                                                } }, { children: "Compare your deduction patterns across multiple years to identify trends." }))] })), _jsxs("div", Object.assign({ style: {
                                                            backgroundColor: "white",
                                                            border: "1px solid #1E40AF1A",
                                                            borderRadius: "0.75rem",
                                                            padding: "1.25rem 1rem",
                                                        } }, { children: [_jsxs("div", Object.assign({ style: {
                                                                    display: "flex",
                                                                    alignItems: "center",
                                                                    justifyContent: "space-between",
                                                                    gap: "2.5rem",
                                                                    marginBottom: "1.25rem",
                                                                } }, { children: [_jsx("h3", Object.assign({ style: {
                                                                            color: "#042567",
                                                                            fontSize: "1.125rem",
                                                                            fontWeight: 500,
                                                                            margin: 0,
                                                                        } }, { children: "Revenue" })), _jsxs("select", Object.assign({ value: view, onChange: (e) => setView(e.target.value), style: {
                                                                            border: "1px solid #1E40AF1A",
                                                                            fontSize: "0.875rem",
                                                                            backgroundColor: "transparent",
                                                                            borderRadius: "9999px",
                                                                            padding: "0.25rem 0.5rem",
                                                                            outline: "none",
                                                                        } }, { children: [_jsx("option", { children: "Yearly" }), _jsx("option", { children: "Monthly" })] }))] })), _jsx("div", { children: _jsx("canvas", { ref: chartRef, height: "100" }) })] })), _jsxs("div", Object.assign({ style: { marginTop: "1rem" } }, { children: [_jsx("h3", Object.assign({ style: {
                                                                    color: "#042567",
                                                                    fontSize: "1.125rem",
                                                                    fontWeight: 500,
                                                                    marginBottom: "0.5rem",
                                                                    marginTop: 0,
                                                                } }, { children: "Key Observations" })), _jsxs("ul", Object.assign({ style: {
                                                                    margin: "0 0 0 20px",
                                                                    padding: 0,
                                                                    listStyleType: "disc",
                                                                } }, { children: [_jsxs("li", Object.assign({ style: {
                                                                            color: "#526282",
                                                                            fontSize: "1rem",
                                                                            fontWeight: 400,
                                                                        } }, { children: ["Your deductions have increased by 33% from 2021 to ", year] })), _jsx("li", Object.assign({ style: {
                                                                            color: "#526282",
                                                                            fontSize: "1rem",
                                                                            fontWeight: 400,
                                                                        } }, { children: "You're claiming 65% of potential deductible expenses" })), _jsx("li", Object.assign({ style: {
                                                                            color: "#526282",
                                                                            fontSize: "1rem",
                                                                            fontWeight: 400,
                                                                        } }, { children: "Office expenses show the most consistent growth year-over-year" }))] }))] }))] })) }))] })))] })) })) }))) }))] }));
};
export default HistoricalAnalysisWidget;
